From: Stained Glass OS <ke7oxh@gmail.com>
Subject: [PATCH] msvcp60: Export the complex number functions under their VC6 names.

Fast mode for msvcp60: VC6's complex exports were spec stubs although the
msvcp90 sources compile the complex code for msvcp60 too. Its names differ
in mangling (non-template functions, _Complex_base<T> without the C
complex type, _Ctr for _Ctraits, the template arguments of the stream
operators, x64 back-reference variants); about 375 entries now name the
same functions as msvcp90's, matched by demangled signature. The compound
assignments that msvcp60 exports as functions of two complex arguments get
cdecl wrappers of the member functions. Left: operator>> for complex
(msvcp60's own ios.c has no complex reading), the wide-stream operators
(not implemented for any version) and function-local constants.
---
 dlls/msvcp60/msvcp60.spec | 726 +++++++++++++++++++++++-----------------------
 dlls/msvcp90/math.c       |  19 ++
 2 files changed, 382 insertions(+), 363 deletions(-)

diff --git a/dlls/msvcp60/msvcp60.spec b/dlls/msvcp60/msvcp60.spec
index 3167d06..b192943 100644
--- a/dlls/msvcp60/msvcp60.spec
+++ b/dlls/msvcp60/msvcp60.spec
@@ -44,12 +44,12 @@
 @ stub -arch=win64 ??$?6DU?$char_traits@D@std@@@std@@YAAEAV?$basic_ostream@DU?$char_traits@D@std@@@0@AEAV10@PEBD@Z
 @ stub -arch=win32 ??$?6DU?$char_traits@D@std@@@std@@YAAAV?$basic_ostream@DU?$char_traits@D@std@@@0@AAV10@PBE@Z
 @ stub -arch=win64 ??$?6DU?$char_traits@D@std@@@std@@YAAEAV?$basic_ostream@DU?$char_traits@D@std@@@0@AEAV10@PEBE@Z
-@ stub -arch=win32 ??$?6DU?$char_traits@D@std@@M@std@@YAAAV?$basic_ostream@DU?$char_traits@D@std@@@0@AAV10@ABV?$complex@M@0@@Z
-@ stub -arch=win64 ??$?6DU?$char_traits@D@std@@M@std@@YAAEAV?$basic_ostream@DU?$char_traits@D@std@@@0@AEAV10@AEBV?$complex@M@0@@Z
-@ stub -arch=win32 ??$?6DU?$char_traits@D@std@@N@std@@YAAAV?$basic_ostream@DU?$char_traits@D@std@@@0@AAV10@ABV?$complex@N@0@@Z
-@ stub -arch=win64 ??$?6DU?$char_traits@D@std@@N@std@@YAAEAV?$basic_ostream@DU?$char_traits@D@std@@@0@AEAV10@AEBV?$complex@N@0@@Z
-@ stub -arch=win32 ??$?6DU?$char_traits@D@std@@O@std@@YAAAV?$basic_ostream@DU?$char_traits@D@std@@@0@AAV10@ABV?$complex@O@0@@Z
-@ stub -arch=win64 ??$?6DU?$char_traits@D@std@@O@std@@YAAEAV?$basic_ostream@DU?$char_traits@D@std@@@0@AEAV10@AEBV?$complex@O@0@@Z
+@ cdecl -arch=win32 ??$?6DU?$char_traits@D@std@@M@std@@YAAAV?$basic_ostream@DU?$char_traits@D@std@@@0@AAV10@ABV?$complex@M@0@@Z(ptr ptr) basic_ostream_char_print_complex_float
+@ cdecl -arch=win64 ??$?6DU?$char_traits@D@std@@M@std@@YAAEAV?$basic_ostream@DU?$char_traits@D@std@@@0@AEAV10@AEBV?$complex@M@0@@Z(ptr ptr) basic_ostream_char_print_complex_float
+@ cdecl -arch=win32 ??$?6DU?$char_traits@D@std@@N@std@@YAAAV?$basic_ostream@DU?$char_traits@D@std@@@0@AAV10@ABV?$complex@N@0@@Z(ptr ptr) basic_ostream_char_print_complex_double
+@ cdecl -arch=win64 ??$?6DU?$char_traits@D@std@@N@std@@YAAEAV?$basic_ostream@DU?$char_traits@D@std@@@0@AEAV10@AEBV?$complex@N@0@@Z(ptr ptr) basic_ostream_char_print_complex_double
+@ cdecl -arch=win32 ??$?6DU?$char_traits@D@std@@O@std@@YAAAV?$basic_ostream@DU?$char_traits@D@std@@@0@AAV10@ABV?$complex@O@0@@Z(ptr ptr) basic_ostream_char_print_complex_ldouble
+@ cdecl -arch=win64 ??$?6DU?$char_traits@D@std@@O@std@@YAAEAV?$basic_ostream@DU?$char_traits@D@std@@@0@AEAV10@AEBV?$complex@O@0@@Z(ptr ptr) basic_ostream_char_print_complex_ldouble
 @ cdecl -arch=win32 ??$?6DU?$char_traits@D@std@@V?$allocator@D@1@@std@@YAAAV?$basic_ostream@DU?$char_traits@D@std@@@0@AAV10@ABV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@0@@Z(ptr ptr) basic_ostream_char_print_bstr
 @ cdecl -arch=win64 ??$?6DU?$char_traits@D@std@@V?$allocator@D@1@@std@@YAAEAV?$basic_ostream@DU?$char_traits@D@std@@@0@AEAV10@AEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@0@@Z(ptr ptr) basic_ostream_char_print_bstr
 @ cdecl -arch=win32 ??$?6GU?$char_traits@G@std@@@std@@YAAAV?$basic_ostream@GU?$char_traits@G@std@@@0@AAV10@G@Z(ptr long) basic_ostream_short_print_ch
@@ -278,93 +278,93 @@
 @ cdecl -arch=win64 ??$?PGU?$char_traits@G@std@@V?$allocator@G@1@@std@@YA_NAEBV?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@0@PEBG@Z(ptr ptr) MSVCP_basic_string_wchar_geq_bstr_cstr
 @ cdecl -arch=win32 ??$?PGU?$char_traits@G@std@@V?$allocator@G@1@@std@@YA_NPBGABV?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@0@@Z(ptr ptr) MSVCP_basic_string_wchar_geq_cstr_bstr
 @ cdecl -arch=win64 ??$?PGU?$char_traits@G@std@@V?$allocator@G@1@@std@@YA_NPEBGAEBV?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@0@@Z(ptr ptr) MSVCP_basic_string_wchar_geq_cstr_bstr
-@ stub -arch=win32 ??$?XMM@std@@YAAAV?$complex@M@0@AAV10@ABV10@@Z
-@ stub -arch=win64 ??$?XMM@std@@YAAEAV?$complex@M@0@AEAV10@AEBV10@@Z
-@ stub -arch=win32 ??$?XNN@std@@YAAAV?$complex@N@0@AAV10@ABV10@@Z
-@ stub -arch=win64 ??$?XNN@std@@YAAEAV?$complex@N@0@AEAV10@AEBV10@@Z
-@ stub -arch=win32 ??$?XOO@std@@YAAAV?$complex@O@0@AAV10@ABV10@@Z
-@ stub -arch=win64 ??$?XOO@std@@YAAEAV?$complex@O@0@AEAV10@AEBV10@@Z
-@ stub -arch=win32 ??$?YMM@std@@YAAAV?$complex@M@0@AAV10@ABV10@@Z
-@ stub -arch=win64 ??$?YMM@std@@YAAEAV?$complex@M@0@AEAV10@AEBV10@@Z
-@ stub -arch=win32 ??$?YNN@std@@YAAAV?$complex@N@0@AAV10@ABV10@@Z
-@ stub -arch=win64 ??$?YNN@std@@YAAEAV?$complex@N@0@AEAV10@AEBV10@@Z
-@ stub -arch=win32 ??$?YOO@std@@YAAAV?$complex@O@0@AAV10@ABV10@@Z
-@ stub -arch=win64 ??$?YOO@std@@YAAEAV?$complex@O@0@AEAV10@AEBV10@@Z
-@ stub -arch=win32 ??$?ZMM@std@@YAAAV?$complex@M@0@AAV10@ABV10@@Z
-@ stub -arch=win64 ??$?ZMM@std@@YAAEAV?$complex@M@0@AEAV10@AEBV10@@Z
-@ stub -arch=win32 ??$?ZNN@std@@YAAAV?$complex@N@0@AAV10@ABV10@@Z
-@ stub -arch=win64 ??$?ZNN@std@@YAAEAV?$complex@N@0@AEAV10@AEBV10@@Z
-@ stub -arch=win32 ??$?ZOO@std@@YAAAV?$complex@O@0@AAV10@ABV10@@Z
-@ stub -arch=win64 ??$?ZOO@std@@YAAEAV?$complex@O@0@AEAV10@AEBV10@@Z
-@ stub -arch=win32 ??$?_0MM@std@@YAAAV?$complex@M@0@AAV10@ABV10@@Z
-@ stub -arch=win64 ??$?_0MM@std@@YAAEAV?$complex@M@0@AEAV10@AEBV10@@Z
-@ stub -arch=win32 ??$?_0NN@std@@YAAAV?$complex@N@0@AAV10@ABV10@@Z
-@ stub -arch=win64 ??$?_0NN@std@@YAAEAV?$complex@N@0@AEAV10@AEBV10@@Z
-@ stub -arch=win32 ??$?_0OO@std@@YAAAV?$complex@O@0@AAV10@ABV10@@Z
-@ stub -arch=win64 ??$?_0OO@std@@YAAEAV?$complex@O@0@AEAV10@AEBV10@@Z
+@ cdecl -arch=win32 ??$?XMM@std@@YAAAV?$complex@M@0@AAV10@ABV10@@Z(ptr ptr) complex_float_mult_assign_tmpl
+@ cdecl -arch=win64 ??$?XMM@std@@YAAEAV?$complex@M@0@AEAV10@AEBV10@@Z(ptr ptr) complex_float_mult_assign_tmpl
+@ cdecl -arch=win32 ??$?XNN@std@@YAAAV?$complex@N@0@AAV10@ABV10@@Z(ptr ptr) complex_double_mult_assign_tmpl
+@ cdecl -arch=win64 ??$?XNN@std@@YAAEAV?$complex@N@0@AEAV10@AEBV10@@Z(ptr ptr) complex_double_mult_assign_tmpl
+@ cdecl -arch=win32 ??$?XOO@std@@YAAAV?$complex@O@0@AAV10@ABV10@@Z(ptr ptr) complex_double_mult_assign_tmpl
+@ cdecl -arch=win64 ??$?XOO@std@@YAAEAV?$complex@O@0@AEAV10@AEBV10@@Z(ptr ptr) complex_double_mult_assign_tmpl
+@ cdecl -arch=win32 ??$?YMM@std@@YAAAV?$complex@M@0@AAV10@ABV10@@Z(ptr ptr) complex_float_add_assign_tmpl
+@ cdecl -arch=win64 ??$?YMM@std@@YAAEAV?$complex@M@0@AEAV10@AEBV10@@Z(ptr ptr) complex_float_add_assign_tmpl
+@ cdecl -arch=win32 ??$?YNN@std@@YAAAV?$complex@N@0@AAV10@ABV10@@Z(ptr ptr) complex_double_add_assign_tmpl
+@ cdecl -arch=win64 ??$?YNN@std@@YAAEAV?$complex@N@0@AEAV10@AEBV10@@Z(ptr ptr) complex_double_add_assign_tmpl
+@ cdecl -arch=win32 ??$?YOO@std@@YAAAV?$complex@O@0@AAV10@ABV10@@Z(ptr ptr) complex_double_add_assign_tmpl
+@ cdecl -arch=win64 ??$?YOO@std@@YAAEAV?$complex@O@0@AEAV10@AEBV10@@Z(ptr ptr) complex_double_add_assign_tmpl
+@ cdecl -arch=win32 ??$?ZMM@std@@YAAAV?$complex@M@0@AAV10@ABV10@@Z(ptr ptr) complex_float_sub_assign_tmpl
+@ cdecl -arch=win64 ??$?ZMM@std@@YAAEAV?$complex@M@0@AEAV10@AEBV10@@Z(ptr ptr) complex_float_sub_assign_tmpl
+@ cdecl -arch=win32 ??$?ZNN@std@@YAAAV?$complex@N@0@AAV10@ABV10@@Z(ptr ptr) complex_double_sub_assign_tmpl
+@ cdecl -arch=win64 ??$?ZNN@std@@YAAEAV?$complex@N@0@AEAV10@AEBV10@@Z(ptr ptr) complex_double_sub_assign_tmpl
+@ cdecl -arch=win32 ??$?ZOO@std@@YAAAV?$complex@O@0@AAV10@ABV10@@Z(ptr ptr) complex_double_sub_assign_tmpl
+@ cdecl -arch=win64 ??$?ZOO@std@@YAAEAV?$complex@O@0@AEAV10@AEBV10@@Z(ptr ptr) complex_double_sub_assign_tmpl
+@ cdecl -arch=win32 ??$?_0MM@std@@YAAAV?$complex@M@0@AAV10@ABV10@@Z(ptr ptr) complex_float_div_assign_tmpl
+@ cdecl -arch=win64 ??$?_0MM@std@@YAAEAV?$complex@M@0@AEAV10@AEBV10@@Z(ptr ptr) complex_float_div_assign_tmpl
+@ cdecl -arch=win32 ??$?_0NN@std@@YAAAV?$complex@N@0@AAV10@ABV10@@Z(ptr ptr) complex_double_div_assign_tmpl
+@ cdecl -arch=win64 ??$?_0NN@std@@YAAEAV?$complex@N@0@AEAV10@AEBV10@@Z(ptr ptr) complex_double_div_assign_tmpl
+@ cdecl -arch=win32 ??$?_0OO@std@@YAAAV?$complex@O@0@AAV10@ABV10@@Z(ptr ptr) complex_double_div_assign_tmpl
+@ cdecl -arch=win64 ??$?_0OO@std@@YAAEAV?$complex@O@0@AEAV10@AEBV10@@Z(ptr ptr) complex_double_div_assign_tmpl
 @ cdecl -arch=win32 ??$_Fabs@M@std@@YAMABV?$complex@M@0@PAH@Z(ptr ptr) complex_float__Fabs
 @ cdecl -arch=win64 ??$_Fabs@M@std@@YAMAEBV?$complex@M@0@PEAH@Z(ptr ptr) complex_float__Fabs
-@ stub -arch=win64 ??$_Fabs@M@std@@YAMAEBV?$complex@M@1@PEAH@Z
+@ cdecl -arch=win64 ??$_Fabs@M@std@@YAMAEBV?$complex@M@1@PEAH@Z(ptr ptr) complex_float__Fabs
 @ cdecl -arch=win32 ??$_Fabs@N@std@@YANABV?$complex@N@0@PAH@Z(ptr ptr) complex_double__Fabs
 @ cdecl -arch=win64 ??$_Fabs@N@std@@YANAEBV?$complex@N@0@PEAH@Z(ptr ptr) complex_double__Fabs
-@ stub -arch=win64 ??$_Fabs@N@std@@YANAEBV?$complex@N@1@PEAH@Z
+@ cdecl -arch=win64 ??$_Fabs@N@std@@YANAEBV?$complex@N@1@PEAH@Z(ptr ptr) complex_double__Fabs
 @ cdecl -arch=win32 ??$_Fabs@O@std@@YAOABV?$complex@O@0@PAH@Z(ptr ptr) complex_double__Fabs
 @ cdecl -arch=win64 ??$_Fabs@O@std@@YAOAEBV?$complex@O@0@PEAH@Z(ptr ptr) complex_double__Fabs
-@ stub -arch=win64 ??$_Fabs@O@std@@YAOAEBV?$complex@O@1@PEAH@Z
+@ cdecl -arch=win64 ??$_Fabs@O@std@@YAOAEBV?$complex@O@1@PEAH@Z(ptr ptr) complex_double__Fabs
 @ cdecl -arch=win32 ??$abs@M@std@@YAMABV?$complex@M@0@@Z(ptr) complex_float_abs
 @ cdecl -arch=win64 ??$abs@M@std@@YAMAEBV?$complex@M@0@@Z(ptr) complex_float_abs
-@ stub -arch=win64 ??$abs@M@std@@YAMAEBV?$complex@M@1@@Z
+@ cdecl -arch=win64 ??$abs@M@std@@YAMAEBV?$complex@M@1@@Z(ptr) complex_float_abs
 @ cdecl -arch=win32 ??$abs@N@std@@YANABV?$complex@N@0@@Z(ptr) complex_double_abs
 @ cdecl -arch=win64 ??$abs@N@std@@YANAEBV?$complex@N@0@@Z(ptr) complex_double_abs
-@ stub -arch=win64 ??$abs@N@std@@YANAEBV?$complex@N@1@@Z
+@ cdecl -arch=win64 ??$abs@N@std@@YANAEBV?$complex@N@1@@Z(ptr) complex_double_abs
 @ cdecl -arch=win32 ??$abs@O@std@@YAOABV?$complex@O@0@@Z(ptr) complex_double_abs
 @ cdecl -arch=win64 ??$abs@O@std@@YAOAEBV?$complex@O@0@@Z(ptr) complex_double_abs
-@ stub -arch=win64 ??$abs@O@std@@YAOAEBV?$complex@O@1@@Z
+@ cdecl -arch=win64 ??$abs@O@std@@YAOAEBV?$complex@O@1@@Z(ptr) complex_double_abs
 @ cdecl -arch=win32 ??$arg@M@std@@YAMABV?$complex@M@0@@Z(ptr) complex_float_arg
 @ cdecl -arch=win64 ??$arg@M@std@@YAMAEBV?$complex@M@0@@Z(ptr) complex_float_arg
-@ stub -arch=win64 ??$arg@M@std@@YAMAEBV?$complex@M@1@@Z
+@ cdecl -arch=win64 ??$arg@M@std@@YAMAEBV?$complex@M@1@@Z(ptr) complex_float_arg
 @ cdecl -arch=win32 ??$arg@N@std@@YANABV?$complex@N@0@@Z(ptr) complex_double_arg
 @ cdecl -arch=win64 ??$arg@N@std@@YANAEBV?$complex@N@0@@Z(ptr) complex_double_arg
-@ stub -arch=win64 ??$arg@N@std@@YANAEBV?$complex@N@1@@Z
+@ cdecl -arch=win64 ??$arg@N@std@@YANAEBV?$complex@N@1@@Z(ptr) complex_double_arg
 @ cdecl -arch=win32 ??$arg@O@std@@YAOABV?$complex@O@0@@Z(ptr) complex_double_arg
 @ cdecl -arch=win64 ??$arg@O@std@@YAOAEBV?$complex@O@0@@Z(ptr) complex_double_arg
-@ stub -arch=win64 ??$arg@O@std@@YAOAEBV?$complex@O@1@@Z
+@ cdecl -arch=win64 ??$arg@O@std@@YAOAEBV?$complex@O@1@@Z(ptr) complex_double_arg
 @ cdecl -arch=win32 ??$conj@M@std@@YA?AV?$complex@M@0@ABV10@@Z(ptr ptr) complex_float_conj
 @ cdecl -arch=win64 ??$conj@M@std@@YA?AV?$complex@M@0@AEBV10@@Z(ptr ptr) complex_float_conj
-@ stub -arch=win64 ??$conj@M@std@@YA?AV?$complex@M@1@AEBV21@@Z
+@ cdecl -arch=win64 ??$conj@M@std@@YA?AV?$complex@M@1@AEBV21@@Z(ptr ptr) complex_float_conj
 @ cdecl -arch=win32 ??$conj@N@std@@YA?AV?$complex@N@0@ABV10@@Z(ptr ptr) complex_double_conj
 @ cdecl -arch=win64 ??$conj@N@std@@YA?AV?$complex@N@0@AEBV10@@Z(ptr ptr) complex_double_conj
-@ stub -arch=win64 ??$conj@N@std@@YA?AV?$complex@N@1@AEBV21@@Z
+@ cdecl -arch=win64 ??$conj@N@std@@YA?AV?$complex@N@1@AEBV21@@Z(ptr ptr) complex_double_conj
 @ cdecl -arch=win32 ??$conj@O@std@@YA?AV?$complex@O@0@ABV10@@Z(ptr ptr) complex_double_conj
 @ cdecl -arch=win64 ??$conj@O@std@@YA?AV?$complex@O@0@AEBV10@@Z(ptr ptr) complex_double_conj
-@ stub -arch=win64 ??$conj@O@std@@YA?AV?$complex@O@1@AEBV21@@Z
+@ cdecl -arch=win64 ??$conj@O@std@@YA?AV?$complex@O@1@AEBV21@@Z(ptr ptr) complex_double_conj
 @ cdecl -arch=win32 ??$cos@M@std@@YA?AV?$complex@M@0@ABV10@@Z(ptr ptr) complex_float_cos
 @ cdecl -arch=win64 ??$cos@M@std@@YA?AV?$complex@M@0@AEBV10@@Z(ptr ptr) complex_float_cos
-@ stub -arch=win64 ??$cos@M@std@@YA?AV?$complex@M@1@AEBV21@@Z
+@ cdecl -arch=win64 ??$cos@M@std@@YA?AV?$complex@M@1@AEBV21@@Z(ptr ptr) complex_float_cos
 @ cdecl -arch=win32 ??$cos@N@std@@YA?AV?$complex@N@0@ABV10@@Z(ptr ptr) complex_double_cos
 @ cdecl -arch=win64 ??$cos@N@std@@YA?AV?$complex@N@0@AEBV10@@Z(ptr ptr) complex_double_cos
-@ stub -arch=win64 ??$cos@N@std@@YA?AV?$complex@N@1@AEBV21@@Z
+@ cdecl -arch=win64 ??$cos@N@std@@YA?AV?$complex@N@1@AEBV21@@Z(ptr ptr) complex_double_cos
 @ cdecl -arch=win32 ??$cos@O@std@@YA?AV?$complex@O@0@ABV10@@Z(ptr ptr) complex_double_cos
 @ cdecl -arch=win64 ??$cos@O@std@@YA?AV?$complex@O@0@AEBV10@@Z(ptr ptr) complex_double_cos
-@ stub -arch=win64 ??$cos@O@std@@YA?AV?$complex@O@1@AEBV21@@Z
+@ cdecl -arch=win64 ??$cos@O@std@@YA?AV?$complex@O@1@AEBV21@@Z(ptr ptr) complex_double_cos
 @ cdecl -arch=win32 ??$cosh@M@std@@YA?AV?$complex@M@0@ABV10@@Z(ptr ptr) complex_float_cosh
 @ cdecl -arch=win64 ??$cosh@M@std@@YA?AV?$complex@M@0@AEBV10@@Z(ptr ptr) complex_float_cosh
-@ stub -arch=win64 ??$cosh@M@std@@YA?AV?$complex@M@1@AEBV21@@Z
+@ cdecl -arch=win64 ??$cosh@M@std@@YA?AV?$complex@M@1@AEBV21@@Z(ptr ptr) complex_float_cosh
 @ cdecl -arch=win32 ??$cosh@N@std@@YA?AV?$complex@N@0@ABV10@@Z(ptr ptr) complex_double_cosh
 @ cdecl -arch=win64 ??$cosh@N@std@@YA?AV?$complex@N@0@AEBV10@@Z(ptr ptr) complex_double_cosh
-@ stub -arch=win64 ??$cosh@N@std@@YA?AV?$complex@N@1@AEBV21@@Z
+@ cdecl -arch=win64 ??$cosh@N@std@@YA?AV?$complex@N@1@AEBV21@@Z(ptr ptr) complex_double_cosh
 @ cdecl -arch=win32 ??$cosh@O@std@@YA?AV?$complex@O@0@ABV10@@Z(ptr ptr) complex_double_cosh
 @ cdecl -arch=win64 ??$cosh@O@std@@YA?AV?$complex@O@0@AEBV10@@Z(ptr ptr) complex_double_cosh
-@ stub -arch=win64 ??$cosh@O@std@@YA?AV?$complex@O@1@AEBV21@@Z
+@ cdecl -arch=win64 ??$cosh@O@std@@YA?AV?$complex@O@1@AEBV21@@Z(ptr ptr) complex_double_cosh
 @ cdecl -arch=win32 ??$exp@M@std@@YA?AV?$complex@M@0@ABV10@@Z(ptr ptr) complex_float_exp
 @ cdecl -arch=win64 ??$exp@M@std@@YA?AV?$complex@M@0@AEBV10@@Z(ptr ptr) complex_float_exp
-@ stub -arch=win64 ??$exp@M@std@@YA?AV?$complex@M@1@AEBV21@@Z
+@ cdecl -arch=win64 ??$exp@M@std@@YA?AV?$complex@M@1@AEBV21@@Z(ptr ptr) complex_float_exp
 @ cdecl -arch=win32 ??$exp@N@std@@YA?AV?$complex@N@0@ABV10@@Z(ptr ptr) complex_double_exp
 @ cdecl -arch=win64 ??$exp@N@std@@YA?AV?$complex@N@0@AEBV10@@Z(ptr ptr) complex_double_exp
-@ stub -arch=win64 ??$exp@N@std@@YA?AV?$complex@N@1@AEBV21@@Z
+@ cdecl -arch=win64 ??$exp@N@std@@YA?AV?$complex@N@1@AEBV21@@Z(ptr ptr) complex_double_exp
 @ cdecl -arch=win32 ??$exp@O@std@@YA?AV?$complex@O@0@ABV10@@Z(ptr ptr) complex_double_exp
 @ cdecl -arch=win64 ??$exp@O@std@@YA?AV?$complex@O@0@AEBV10@@Z(ptr ptr) complex_double_exp
-@ stub -arch=win64 ??$exp@O@std@@YA?AV?$complex@O@1@AEBV21@@Z
+@ cdecl -arch=win64 ??$exp@O@std@@YA?AV?$complex@O@1@AEBV21@@Z(ptr ptr) complex_double_exp
 @ cdecl -arch=win32 ??$getline@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@YAAAV?$basic_istream@DU?$char_traits@D@std@@@0@AAV10@AAV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@0@@Z(ptr ptr) basic_istream_char_getline_bstr
 @ cdecl -arch=win64 ??$getline@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@YAAEAV?$basic_istream@DU?$char_traits@D@std@@@0@AEAV10@AEAV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@0@@Z(ptr ptr) basic_istream_char_getline_bstr
 @ stub -arch=win64 ??$getline@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@YAAEAV?$basic_istream@DU?$char_traits@D@std@@@1@AEAV21@AEAV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@1@D@Z
@@ -379,139 +379,139 @@
 @ stub -arch=win64 ??$getline@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@YAAEAV?$basic_istream@GU?$char_traits@G@std@@@1@AEAV21@AEAV?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@1@G@Z
 @ cdecl -arch=win32 ??$imag@M@std@@YAMABV?$complex@M@0@@Z(ptr) complex_float_imag
 @ cdecl -arch=win64 ??$imag@M@std@@YAMAEBV?$complex@M@0@@Z(ptr) complex_float_imag
-@ stub -arch=win64 ??$imag@M@std@@YAMAEBV?$complex@M@1@@Z
+@ cdecl -arch=win64 ??$imag@M@std@@YAMAEBV?$complex@M@1@@Z(ptr) complex_float_imag
 @ cdecl -arch=win32 ??$imag@N@std@@YANABV?$complex@N@0@@Z(ptr) complex_double_imag
 @ cdecl -arch=win64 ??$imag@N@std@@YANAEBV?$complex@N@0@@Z(ptr) complex_double_imag
-@ stub -arch=win64 ??$imag@N@std@@YANAEBV?$complex@N@1@@Z
+@ cdecl -arch=win64 ??$imag@N@std@@YANAEBV?$complex@N@1@@Z(ptr) complex_double_imag
 @ cdecl -arch=win32 ??$imag@O@std@@YAOABV?$complex@O@0@@Z(ptr) complex_double_imag
 @ cdecl -arch=win64 ??$imag@O@std@@YAOAEBV?$complex@O@0@@Z(ptr) complex_double_imag
-@ stub -arch=win64 ??$imag@O@std@@YAOAEBV?$complex@O@1@@Z
+@ cdecl -arch=win64 ??$imag@O@std@@YAOAEBV?$complex@O@1@@Z(ptr) complex_double_imag
 @ cdecl -arch=win32 ??$log10@M@std@@YA?AV?$complex@M@0@ABV10@@Z(ptr ptr) complex_float_log10
 @ cdecl -arch=win64 ??$log10@M@std@@YA?AV?$complex@M@0@AEBV10@@Z(ptr ptr) complex_float_log10
-@ stub -arch=win64 ??$log10@M@std@@YA?AV?$complex@M@1@AEBV21@@Z
+@ cdecl -arch=win64 ??$log10@M@std@@YA?AV?$complex@M@1@AEBV21@@Z(ptr ptr) complex_float_log10
 @ cdecl -arch=win32 ??$log10@N@std@@YA?AV?$complex@N@0@ABV10@@Z(ptr ptr) complex_double_log10
 @ cdecl -arch=win64 ??$log10@N@std@@YA?AV?$complex@N@0@AEBV10@@Z(ptr ptr) complex_double_log10
-@ stub -arch=win64 ??$log10@N@std@@YA?AV?$complex@N@1@AEBV21@@Z
+@ cdecl -arch=win64 ??$log10@N@std@@YA?AV?$complex@N@1@AEBV21@@Z(ptr ptr) complex_double_log10
 @ cdecl -arch=win32 ??$log10@O@std@@YA?AV?$complex@O@0@ABV10@@Z(ptr ptr) complex_double_log10
 @ cdecl -arch=win64 ??$log10@O@std@@YA?AV?$complex@O@0@AEBV10@@Z(ptr ptr) complex_double_log10
-@ stub -arch=win64 ??$log10@O@std@@YA?AV?$complex@O@1@AEBV21@@Z
+@ cdecl -arch=win64 ??$log10@O@std@@YA?AV?$complex@O@1@AEBV21@@Z(ptr ptr) complex_double_log10
 @ cdecl -arch=win32 ??$log@M@std@@YA?AV?$complex@M@0@ABV10@@Z(ptr ptr) complex_float_log
 @ cdecl -arch=win64 ??$log@M@std@@YA?AV?$complex@M@0@AEBV10@@Z(ptr ptr) complex_float_log
-@ stub -arch=win64 ??$log@M@std@@YA?AV?$complex@M@1@AEBV21@@Z
+@ cdecl -arch=win64 ??$log@M@std@@YA?AV?$complex@M@1@AEBV21@@Z(ptr ptr) complex_float_log
 @ cdecl -arch=win32 ??$log@N@std@@YA?AV?$complex@N@0@ABV10@@Z(ptr ptr) complex_double_log
 @ cdecl -arch=win64 ??$log@N@std@@YA?AV?$complex@N@0@AEBV10@@Z(ptr ptr) complex_double_log
-@ stub -arch=win64 ??$log@N@std@@YA?AV?$complex@N@1@AEBV21@@Z
+@ cdecl -arch=win64 ??$log@N@std@@YA?AV?$complex@N@1@AEBV21@@Z(ptr ptr) complex_double_log
 @ cdecl -arch=win32 ??$log@O@std@@YA?AV?$complex@O@0@ABV10@@Z(ptr ptr) complex_double_log
 @ cdecl -arch=win64 ??$log@O@std@@YA?AV?$complex@O@0@AEBV10@@Z(ptr ptr) complex_double_log
-@ stub -arch=win64 ??$log@O@std@@YA?AV?$complex@O@1@AEBV21@@Z
+@ cdecl -arch=win64 ??$log@O@std@@YA?AV?$complex@O@1@AEBV21@@Z(ptr ptr) complex_double_log
 @ cdecl -arch=win32 ??$norm@M@std@@YAMABV?$complex@M@0@@Z(ptr) complex_float_norm
 @ cdecl -arch=win64 ??$norm@M@std@@YAMAEBV?$complex@M@0@@Z(ptr) complex_float_norm
-@ stub -arch=win64 ??$norm@M@std@@YAMAEBV?$complex@M@1@@Z
+@ cdecl -arch=win64 ??$norm@M@std@@YAMAEBV?$complex@M@1@@Z(ptr) complex_float_norm
 @ cdecl -arch=win32 ??$norm@N@std@@YANABV?$complex@N@0@@Z(ptr) complex_double_norm
 @ cdecl -arch=win64 ??$norm@N@std@@YANAEBV?$complex@N@0@@Z(ptr) complex_double_norm
-@ stub -arch=win64 ??$norm@N@std@@YANAEBV?$complex@N@1@@Z
+@ cdecl -arch=win64 ??$norm@N@std@@YANAEBV?$complex@N@1@@Z(ptr) complex_double_norm
 @ cdecl -arch=win32 ??$norm@O@std@@YAOABV?$complex@O@0@@Z(ptr) complex_double_norm
 @ cdecl -arch=win64 ??$norm@O@std@@YAOAEBV?$complex@O@0@@Z(ptr) complex_double_norm
-@ stub -arch=win64 ??$norm@O@std@@YAOAEBV?$complex@O@1@@Z
+@ cdecl -arch=win64 ??$norm@O@std@@YAOAEBV?$complex@O@1@@Z(ptr) complex_double_norm
 @ cdecl -arch=win32 ??$polar@M@std@@YA?AV?$complex@M@0@ABM0@Z(ptr ptr ptr) complex_float_polar_theta
 @ cdecl -arch=win64 ??$polar@M@std@@YA?AV?$complex@M@0@AEBM0@Z(ptr ptr ptr) complex_float_polar_theta
-@ stub -arch=win64 ??$polar@M@std@@YA?AV?$complex@M@1@AEBM0@Z
+@ cdecl -arch=win64 ??$polar@M@std@@YA?AV?$complex@M@1@AEBM0@Z(ptr ptr ptr) complex_float_polar_theta
 @ cdecl -arch=win32 ??$polar@M@std@@YA?AV?$complex@M@0@ABM@Z(ptr ptr) complex_float_polar
 @ cdecl -arch=win64 ??$polar@M@std@@YA?AV?$complex@M@0@AEBM@Z(ptr ptr) complex_float_polar
-@ stub -arch=win64 ??$polar@M@std@@YA?AV?$complex@M@1@AEBM@Z
+@ cdecl -arch=win64 ??$polar@M@std@@YA?AV?$complex@M@1@AEBM@Z(ptr ptr) complex_float_polar
 @ cdecl -arch=win32 ??$polar@N@std@@YA?AV?$complex@N@0@ABN0@Z(ptr ptr ptr) complex_double_polar_theta
 @ cdecl -arch=win64 ??$polar@N@std@@YA?AV?$complex@N@0@AEBN0@Z(ptr ptr ptr) complex_double_polar_theta
-@ stub -arch=win64 ??$polar@N@std@@YA?AV?$complex@N@1@AEBN0@Z
+@ cdecl -arch=win64 ??$polar@N@std@@YA?AV?$complex@N@1@AEBN0@Z(ptr ptr ptr) complex_double_polar_theta
 @ cdecl -arch=win32 ??$polar@N@std@@YA?AV?$complex@N@0@ABN@Z(ptr ptr) complex_double_polar
 @ cdecl -arch=win64 ??$polar@N@std@@YA?AV?$complex@N@0@AEBN@Z(ptr ptr) complex_double_polar
-@ stub -arch=win64 ??$polar@N@std@@YA?AV?$complex@N@1@AEBN@Z
+@ cdecl -arch=win64 ??$polar@N@std@@YA?AV?$complex@N@1@AEBN@Z(ptr ptr) complex_double_polar
 @ cdecl -arch=win32 ??$polar@O@std@@YA?AV?$complex@O@0@ABO0@Z(ptr ptr ptr) complex_double_polar_theta
 @ cdecl -arch=win64 ??$polar@O@std@@YA?AV?$complex@O@0@AEBO0@Z(ptr ptr ptr) complex_double_polar_theta
-@ stub -arch=win64 ??$polar@O@std@@YA?AV?$complex@O@1@AEBO0@Z
+@ cdecl -arch=win64 ??$polar@O@std@@YA?AV?$complex@O@1@AEBO0@Z(ptr ptr ptr) complex_double_polar_theta
 @ cdecl -arch=win32 ??$polar@O@std@@YA?AV?$complex@O@0@ABO@Z(ptr ptr) complex_double_polar
 @ cdecl -arch=win64 ??$polar@O@std@@YA?AV?$complex@O@0@AEBO@Z(ptr ptr) complex_double_polar
-@ stub -arch=win64 ??$polar@O@std@@YA?AV?$complex@O@1@AEBO@Z
+@ cdecl -arch=win64 ??$polar@O@std@@YA?AV?$complex@O@1@AEBO@Z(ptr ptr) complex_double_polar
 @ cdecl -arch=win32 ??$pow@M@std@@YA?AV?$complex@M@0@ABMABV10@@Z(ptr ptr ptr) complex_float_pow_fc
 @ cdecl -arch=win64 ??$pow@M@std@@YA?AV?$complex@M@0@AEBMAEBV10@@Z(ptr ptr ptr) complex_float_pow_fc
-@ stub -arch=win64 ??$pow@M@std@@YA?AV?$complex@M@1@AEBMAEBV21@@Z
+@ cdecl -arch=win64 ??$pow@M@std@@YA?AV?$complex@M@1@AEBMAEBV21@@Z(ptr ptr ptr) complex_float_pow_fc
 @ cdecl -arch=win32 ??$pow@M@std@@YA?AV?$complex@M@0@ABV10@0@Z(ptr ptr ptr) complex_float_pow
 @ cdecl -arch=win64 ??$pow@M@std@@YA?AV?$complex@M@0@AEBV10@0@Z(ptr ptr ptr) complex_float_pow
-@ stub -arch=win64 ??$pow@M@std@@YA?AV?$complex@M@1@AEBV21@0@Z
+@ cdecl -arch=win64 ??$pow@M@std@@YA?AV?$complex@M@1@AEBV21@0@Z(ptr ptr ptr) complex_float_pow
 @ cdecl -arch=win32 ??$pow@M@std@@YA?AV?$complex@M@0@ABV10@ABM@Z(ptr ptr ptr) complex_float_pow_cf
 @ cdecl -arch=win64 ??$pow@M@std@@YA?AV?$complex@M@0@AEBV10@AEBM@Z(ptr ptr ptr) complex_float_pow_cf
-@ stub -arch=win64 ??$pow@M@std@@YA?AV?$complex@M@1@AEBV21@AEBM@Z
+@ cdecl -arch=win64 ??$pow@M@std@@YA?AV?$complex@M@1@AEBV21@AEBM@Z(ptr ptr ptr) complex_float_pow_cf
 @ cdecl -arch=win32 ??$pow@M@std@@YA?AV?$complex@M@0@ABV10@H@Z(ptr ptr long) complex_float_pow_ci
 @ cdecl -arch=win64 ??$pow@M@std@@YA?AV?$complex@M@0@AEBV10@H@Z(ptr ptr long) complex_float_pow_ci
-@ stub -arch=win64 ??$pow@M@std@@YA?AV?$complex@M@1@AEBV21@H@Z
+@ cdecl -arch=win64 ??$pow@M@std@@YA?AV?$complex@M@1@AEBV21@H@Z(ptr ptr long) complex_float_pow_ci
 @ cdecl -arch=win32 ??$pow@N@std@@YA?AV?$complex@N@0@ABNABV10@@Z(ptr ptr ptr) complex_double_pow_dc
 @ cdecl -arch=win64 ??$pow@N@std@@YA?AV?$complex@N@0@AEBNAEBV10@@Z(ptr ptr ptr) complex_double_pow_dc
-@ stub -arch=win64 ??$pow@N@std@@YA?AV?$complex@N@1@AEBNAEBV21@@Z
+@ cdecl -arch=win64 ??$pow@N@std@@YA?AV?$complex@N@1@AEBNAEBV21@@Z(ptr ptr ptr) complex_double_pow_dc
 @ cdecl -arch=win32 ??$pow@N@std@@YA?AV?$complex@N@0@ABV10@0@Z(ptr ptr ptr) complex_double_pow
 @ cdecl -arch=win64 ??$pow@N@std@@YA?AV?$complex@N@0@AEBV10@0@Z(ptr ptr ptr) complex_double_pow
-@ stub -arch=win64 ??$pow@N@std@@YA?AV?$complex@N@1@AEBV21@0@Z
+@ cdecl -arch=win64 ??$pow@N@std@@YA?AV?$complex@N@1@AEBV21@0@Z(ptr ptr ptr) complex_double_pow
 @ cdecl -arch=win32 ??$pow@N@std@@YA?AV?$complex@N@0@ABV10@ABN@Z(ptr ptr ptr) complex_double_pow_cd
 @ cdecl -arch=win64 ??$pow@N@std@@YA?AV?$complex@N@0@AEBV10@AEBN@Z(ptr ptr ptr) complex_double_pow_cd
-@ stub -arch=win64 ??$pow@N@std@@YA?AV?$complex@N@1@AEBV21@AEBN@Z
+@ cdecl -arch=win64 ??$pow@N@std@@YA?AV?$complex@N@1@AEBV21@AEBN@Z(ptr ptr ptr) complex_double_pow_cd
 @ cdecl -arch=win32 ??$pow@N@std@@YA?AV?$complex@N@0@ABV10@H@Z(ptr ptr long) complex_double_pow_ci
 @ cdecl -arch=win64 ??$pow@N@std@@YA?AV?$complex@N@0@AEBV10@H@Z(ptr ptr long) complex_double_pow_ci
-@ stub -arch=win64 ??$pow@N@std@@YA?AV?$complex@N@1@AEBV21@H@Z
+@ cdecl -arch=win64 ??$pow@N@std@@YA?AV?$complex@N@1@AEBV21@H@Z(ptr ptr long) complex_double_pow_ci
 @ cdecl -arch=win32 ??$pow@O@std@@YA?AV?$complex@O@0@ABOABV10@@Z(ptr ptr ptr) complex_double_pow_dc
 @ cdecl -arch=win64 ??$pow@O@std@@YA?AV?$complex@O@0@AEBOAEBV10@@Z(ptr ptr ptr) complex_double_pow_dc
-@ stub -arch=win64 ??$pow@O@std@@YA?AV?$complex@O@1@AEBOAEBV21@@Z
+@ cdecl -arch=win64 ??$pow@O@std@@YA?AV?$complex@O@1@AEBOAEBV21@@Z(ptr ptr ptr) complex_double_pow_dc
 @ cdecl -arch=win32 ??$pow@O@std@@YA?AV?$complex@O@0@ABV10@0@Z(ptr ptr ptr) complex_double_pow
 @ cdecl -arch=win64 ??$pow@O@std@@YA?AV?$complex@O@0@AEBV10@0@Z(ptr ptr ptr) complex_double_pow
-@ stub -arch=win64 ??$pow@O@std@@YA?AV?$complex@O@1@AEBV21@0@Z
+@ cdecl -arch=win64 ??$pow@O@std@@YA?AV?$complex@O@1@AEBV21@0@Z(ptr ptr ptr) complex_double_pow
 @ cdecl -arch=win32 ??$pow@O@std@@YA?AV?$complex@O@0@ABV10@ABO@Z(ptr ptr ptr) complex_double_pow_cd
 @ cdecl -arch=win64 ??$pow@O@std@@YA?AV?$complex@O@0@AEBV10@AEBO@Z(ptr ptr ptr) complex_double_pow_cd
-@ stub -arch=win64 ??$pow@O@std@@YA?AV?$complex@O@1@AEBV21@AEBO@Z
+@ cdecl -arch=win64 ??$pow@O@std@@YA?AV?$complex@O@1@AEBV21@AEBO@Z(ptr ptr ptr) complex_double_pow_cd
 @ cdecl -arch=win32 ??$pow@O@std@@YA?AV?$complex@O@0@ABV10@H@Z(ptr ptr long) complex_double_pow_ci
 @ cdecl -arch=win64 ??$pow@O@std@@YA?AV?$complex@O@0@AEBV10@H@Z(ptr ptr long) complex_double_pow_ci
-@ stub -arch=win64 ??$pow@O@std@@YA?AV?$complex@O@1@AEBV21@H@Z
+@ cdecl -arch=win64 ??$pow@O@std@@YA?AV?$complex@O@1@AEBV21@H@Z(ptr ptr long) complex_double_pow_ci
 @ cdecl -arch=win32 ??$real@M@std@@YAMABV?$complex@M@0@@Z(ptr) complex_float_real
 @ cdecl -arch=win64 ??$real@M@std@@YAMAEBV?$complex@M@0@@Z(ptr) complex_float_real
-@ stub -arch=win64 ??$real@M@std@@YAMAEBV?$complex@M@1@@Z
+@ cdecl -arch=win64 ??$real@M@std@@YAMAEBV?$complex@M@1@@Z(ptr) complex_float_real
 @ cdecl -arch=win32 ??$real@N@std@@YANABV?$complex@N@0@@Z(ptr) complex_double_real
 @ cdecl -arch=win64 ??$real@N@std@@YANAEBV?$complex@N@0@@Z(ptr) complex_double_real
-@ stub -arch=win64 ??$real@N@std@@YANAEBV?$complex@N@1@@Z
+@ cdecl -arch=win64 ??$real@N@std@@YANAEBV?$complex@N@1@@Z(ptr) complex_double_real
 @ cdecl -arch=win32 ??$real@O@std@@YAOABV?$complex@O@0@@Z(ptr) complex_double_real
 @ cdecl -arch=win64 ??$real@O@std@@YAOAEBV?$complex@O@0@@Z(ptr) complex_double_real
-@ stub -arch=win64 ??$real@O@std@@YAOAEBV?$complex@O@1@@Z
+@ cdecl -arch=win64 ??$real@O@std@@YAOAEBV?$complex@O@1@@Z(ptr) complex_double_real
 @ cdecl -arch=win32 ??$sin@M@std@@YA?AV?$complex@M@0@ABV10@@Z(ptr ptr) complex_float_sin
 @ cdecl -arch=win64 ??$sin@M@std@@YA?AV?$complex@M@0@AEBV10@@Z(ptr ptr) complex_float_sin
-@ stub -arch=win64 ??$sin@M@std@@YA?AV?$complex@M@1@AEBV21@@Z
+@ cdecl -arch=win64 ??$sin@M@std@@YA?AV?$complex@M@1@AEBV21@@Z(ptr ptr) complex_float_sin
 @ cdecl -arch=win32 ??$sin@N@std@@YA?AV?$complex@N@0@ABV10@@Z(ptr ptr) complex_double_sin
 @ cdecl -arch=win64 ??$sin@N@std@@YA?AV?$complex@N@0@AEBV10@@Z(ptr ptr) complex_double_sin
-@ stub -arch=win64 ??$sin@N@std@@YA?AV?$complex@N@1@AEBV21@@Z
+@ cdecl -arch=win64 ??$sin@N@std@@YA?AV?$complex@N@1@AEBV21@@Z(ptr ptr) complex_double_sin
 @ cdecl -arch=win32 ??$sin@O@std@@YA?AV?$complex@O@0@ABV10@@Z(ptr ptr) complex_double_sin
 @ cdecl -arch=win64 ??$sin@O@std@@YA?AV?$complex@O@0@AEBV10@@Z(ptr ptr) complex_double_sin
-@ stub -arch=win64 ??$sin@O@std@@YA?AV?$complex@O@1@AEBV21@@Z
+@ cdecl -arch=win64 ??$sin@O@std@@YA?AV?$complex@O@1@AEBV21@@Z(ptr ptr) complex_double_sin
 @ cdecl -arch=win32 ??$sinh@M@std@@YA?AV?$complex@M@0@ABV10@@Z(ptr ptr) complex_float_sinh
 @ cdecl -arch=win64 ??$sinh@M@std@@YA?AV?$complex@M@0@AEBV10@@Z(ptr ptr) complex_float_sinh
-@ stub -arch=win64 ??$sinh@M@std@@YA?AV?$complex@M@1@AEBV21@@Z
+@ cdecl -arch=win64 ??$sinh@M@std@@YA?AV?$complex@M@1@AEBV21@@Z(ptr ptr) complex_float_sinh
 @ cdecl -arch=win32 ??$sinh@N@std@@YA?AV?$complex@N@0@ABV10@@Z(ptr ptr) complex_double_sinh
 @ cdecl -arch=win64 ??$sinh@N@std@@YA?AV?$complex@N@0@AEBV10@@Z(ptr ptr) complex_double_sinh
-@ stub -arch=win64 ??$sinh@N@std@@YA?AV?$complex@N@1@AEBV21@@Z
+@ cdecl -arch=win64 ??$sinh@N@std@@YA?AV?$complex@N@1@AEBV21@@Z(ptr ptr) complex_double_sinh
 @ cdecl -arch=win32 ??$sinh@O@std@@YA?AV?$complex@O@0@ABV10@@Z(ptr ptr) complex_double_sinh
 @ cdecl -arch=win64 ??$sinh@O@std@@YA?AV?$complex@O@0@AEBV10@@Z(ptr ptr) complex_double_sinh
-@ stub -arch=win64 ??$sinh@O@std@@YA?AV?$complex@O@1@AEBV21@@Z
+@ cdecl -arch=win64 ??$sinh@O@std@@YA?AV?$complex@O@1@AEBV21@@Z(ptr ptr) complex_double_sinh
 @ cdecl -arch=win32 ??$sqrt@M@std@@YA?AV?$complex@M@0@ABV10@@Z(ptr ptr) complex_float_sqrt
 @ cdecl -arch=win64 ??$sqrt@M@std@@YA?AV?$complex@M@0@AEBV10@@Z(ptr ptr) complex_float_sqrt
-@ stub -arch=win64 ??$sqrt@M@std@@YA?AV?$complex@M@1@AEBV21@@Z
+@ cdecl -arch=win64 ??$sqrt@M@std@@YA?AV?$complex@M@1@AEBV21@@Z(ptr ptr) complex_float_sqrt
 @ cdecl -arch=win32 ??$sqrt@N@std@@YA?AV?$complex@N@0@ABV10@@Z(ptr ptr) complex_double_sqrt
 @ cdecl -arch=win64 ??$sqrt@N@std@@YA?AV?$complex@N@0@AEBV10@@Z(ptr ptr) complex_double_sqrt
-@ stub -arch=win64 ??$sqrt@N@std@@YA?AV?$complex@N@1@AEBV21@@Z
+@ cdecl -arch=win64 ??$sqrt@N@std@@YA?AV?$complex@N@1@AEBV21@@Z(ptr ptr) complex_double_sqrt
 @ cdecl -arch=win32 ??$sqrt@O@std@@YA?AV?$complex@O@0@ABV10@@Z(ptr ptr) complex_double_sqrt
 @ cdecl -arch=win64 ??$sqrt@O@std@@YA?AV?$complex@O@0@AEBV10@@Z(ptr ptr) complex_double_sqrt
-@ stub -arch=win64 ??$sqrt@O@std@@YA?AV?$complex@O@1@AEBV21@@Z
+@ cdecl -arch=win64 ??$sqrt@O@std@@YA?AV?$complex@O@1@AEBV21@@Z(ptr ptr) complex_double_sqrt
 @ stub -arch=arm ??0?$_Complex_base@M@std@@QAA@ABM0@Z
-@ stub -arch=i386 ??0?$_Complex_base@M@std@@QAE@ABM0@Z
-@ stub -arch=win64 ??0?$_Complex_base@M@std@@QEAA@AEBM0@Z
+@ thiscall -arch=i386 ??0?$_Complex_base@M@std@@QAE@ABM0@Z(ptr ptr ptr) complex_float_ctor
+@ cdecl -arch=win64 ??0?$_Complex_base@M@std@@QEAA@AEBM0@Z(ptr ptr ptr) complex_float_ctor
 @ stub -arch=arm ??0?$_Complex_base@N@std@@QAA@ABN0@Z
-@ stub -arch=i386 ??0?$_Complex_base@N@std@@QAE@ABN0@Z
-@ stub -arch=win64 ??0?$_Complex_base@N@std@@QEAA@AEBN0@Z
+@ thiscall -arch=i386 ??0?$_Complex_base@N@std@@QAE@ABN0@Z(ptr ptr ptr) complex_double_ctor
+@ cdecl -arch=win64 ??0?$_Complex_base@N@std@@QEAA@AEBN0@Z(ptr ptr ptr) complex_double_ctor
 @ stub -arch=arm ??0?$_Complex_base@O@std@@QAA@ABO0@Z
-@ stub -arch=i386 ??0?$_Complex_base@O@std@@QAE@ABO0@Z
-@ stub -arch=win64 ??0?$_Complex_base@O@std@@QEAA@AEBO0@Z
+@ thiscall -arch=i386 ??0?$_Complex_base@O@std@@QAE@ABO0@Z(ptr ptr ptr) complex_double_ctor
+@ cdecl -arch=win64 ??0?$_Complex_base@O@std@@QEAA@AEBO0@Z(ptr ptr ptr) complex_double_ctor
 @ cdecl -arch=arm ??0?$_Mpunct@D@std@@QAA@ABV_Locinfo@1@I_N@Z(ptr long long) mpunct_char_ctor_refs
 @ thiscall -arch=i386 ??0?$_Mpunct@D@std@@QAE@ABV_Locinfo@1@I_N@Z(ptr long long) mpunct_char_ctor_refs
 @ cdecl -arch=win64 ??0?$_Mpunct@D@std@@QEAA@AEBV_Locinfo@1@_K_N@Z(ptr long long) mpunct_char_ctor_refs
@@ -1392,23 +1392,23 @@
 @ stub -arch=i386 ??1underflow_error@std@@UAE@XZ
 @ stub -arch=win64 ??1underflow_error@std@@UEAA@XZ
 @ stub -arch=arm ??4?$_Complex_base@M@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4?$_Complex_base@M@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4?$_Complex_base@M@std@@QEAAAEAV01@AEBV01@@Z
+@ thiscall -arch=i386 ??4?$_Complex_base@M@std@@QAEAAV01@ABV01@@Z(ptr ptr) complex_float_assign
+@ cdecl -arch=win64 ??4?$_Complex_base@M@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) complex_float_assign
 @ stub -arch=arm ??4?$_Complex_base@N@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4?$_Complex_base@N@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4?$_Complex_base@N@std@@QEAAAEAV01@AEBV01@@Z
+@ thiscall -arch=i386 ??4?$_Complex_base@N@std@@QAEAAV01@ABV01@@Z(ptr ptr) complex_double_assign
+@ cdecl -arch=win64 ??4?$_Complex_base@N@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) complex_double_assign
 @ stub -arch=arm ??4?$_Complex_base@O@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4?$_Complex_base@O@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4?$_Complex_base@O@std@@QEAAAEAV01@AEBV01@@Z
+@ thiscall -arch=i386 ??4?$_Complex_base@O@std@@QAEAAV01@ABV01@@Z(ptr ptr) complex_double_assign
+@ cdecl -arch=win64 ??4?$_Complex_base@O@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) complex_double_assign
 @ stub -arch=arm ??4?$_Ctr@M@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4?$_Ctr@M@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4?$_Ctr@M@std@@QEAAAEAV01@AEBV01@@Z
+@ thiscall -arch=i386 ??4?$_Ctr@M@std@@QAEAAV01@ABV01@@Z(ptr ptr) std_Ctraits_op_assign
+@ cdecl -arch=win64 ??4?$_Ctr@M@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) std_Ctraits_op_assign
 @ stub -arch=arm ??4?$_Ctr@N@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4?$_Ctr@N@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4?$_Ctr@N@std@@QEAAAEAV01@AEBV01@@Z
+@ thiscall -arch=i386 ??4?$_Ctr@N@std@@QAEAAV01@ABV01@@Z(ptr ptr) std_Ctraits_op_assign
+@ cdecl -arch=win64 ??4?$_Ctr@N@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) std_Ctraits_op_assign
 @ stub -arch=arm ??4?$_Ctr@O@std@@QAAAAV01@ABV01@@Z
-@ stub -arch=i386 ??4?$_Ctr@O@std@@QAEAAV01@ABV01@@Z
-@ stub -arch=win64 ??4?$_Ctr@O@std@@QEAAAEAV01@AEBV01@@Z
+@ thiscall -arch=i386 ??4?$_Ctr@O@std@@QAEAAV01@ABV01@@Z(ptr ptr) std_Ctraits_op_assign
+@ cdecl -arch=win64 ??4?$_Ctr@O@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) std_Ctraits_op_assign
 @ cdecl -arch=arm ??4?$allocator@X@std@@QAAAAV01@ABV01@@Z(ptr ptr) MSVCP_allocator_void_assign
 @ thiscall -arch=i386 ??4?$allocator@X@std@@QAEAAV01@ABV01@@Z(ptr ptr) MSVCP_allocator_void_assign
 @ cdecl -arch=win64 ??4?$allocator@X@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) MSVCP_allocator_void_assign
@@ -2157,30 +2157,30 @@
 @ stub -arch=arm ??Rlocale@std@@QBA_NABV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@1@0@Z
 @ stub -arch=i386 ??Rlocale@std@@QBE_NABV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@1@0@Z
 @ stub -arch=win64 ??Rlocale@std@@QEBA_NAEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@1@0@Z
-@ stub -arch=arm ??X?$_Complex_base@M@std@@QAAAAV01@ABM@Z
-@ stub -arch=i386 ??X?$_Complex_base@M@std@@QAEAAV01@ABM@Z
-@ stub -arch=win64 ??X?$_Complex_base@M@std@@QEAAAEAV01@AEBM@Z
-@ stub -arch=arm ??X?$_Complex_base@N@std@@QAAAAV01@ABN@Z
-@ stub -arch=i386 ??X?$_Complex_base@N@std@@QAEAAV01@ABN@Z
-@ stub -arch=win64 ??X?$_Complex_base@N@std@@QEAAAEAV01@AEBN@Z
-@ stub -arch=arm ??X?$_Complex_base@O@std@@QAAAAV01@ABO@Z
-@ stub -arch=i386 ??X?$_Complex_base@O@std@@QAEAAV01@ABO@Z
-@ stub -arch=win64 ??X?$_Complex_base@O@std@@QEAAAEAV01@AEBO@Z
-@ stub -arch=win32 ??Xstd@@YAAAV?$complex@M@0@AAV10@ABV10@@Z
-@ stub -arch=win64 ??Xstd@@YAAEAV?$complex@M@0@AEAV10@AEBV10@@Z
-@ stub -arch=win32 ??Xstd@@YAAAV?$complex@N@0@AAV10@ABV10@@Z
-@ stub -arch=win64 ??Xstd@@YAAEAV?$complex@N@0@AEAV10@AEBV10@@Z
-@ stub -arch=win32 ??Xstd@@YAAAV?$complex@O@0@AAV10@ABV10@@Z
-@ stub -arch=win64 ??Xstd@@YAAEAV?$complex@O@0@AEAV10@AEBV10@@Z
-@ stub -arch=arm ??Y?$_Complex_base@M@std@@QAAAAV01@ABM@Z
-@ stub -arch=i386 ??Y?$_Complex_base@M@std@@QAEAAV01@ABM@Z
-@ stub -arch=win64 ??Y?$_Complex_base@M@std@@QEAAAEAV01@AEBM@Z
-@ stub -arch=arm ??Y?$_Complex_base@N@std@@QAAAAV01@ABN@Z
-@ stub -arch=i386 ??Y?$_Complex_base@N@std@@QAEAAV01@ABN@Z
-@ stub -arch=win64 ??Y?$_Complex_base@N@std@@QEAAAEAV01@AEBN@Z
-@ stub -arch=arm ??Y?$_Complex_base@O@std@@QAAAAV01@ABO@Z
-@ stub -arch=i386 ??Y?$_Complex_base@O@std@@QAEAAV01@ABO@Z
-@ stub -arch=win64 ??Y?$_Complex_base@O@std@@QEAAAEAV01@AEBO@Z
+@ cdecl -arch=arm ??X?$_Complex_base@M@std@@QAAAAV01@ABM@Z(ptr ptr) complex_float_mult_assign_float
+@ thiscall -arch=i386 ??X?$_Complex_base@M@std@@QAEAAV01@ABM@Z(ptr ptr) complex_float_mult_assign_float
+@ cdecl -arch=win64 ??X?$_Complex_base@M@std@@QEAAAEAV01@AEBM@Z(ptr ptr) complex_float_mult_assign_float
+@ cdecl -arch=arm ??X?$_Complex_base@N@std@@QAAAAV01@ABN@Z(ptr ptr) complex_double_mult_assign_double
+@ thiscall -arch=i386 ??X?$_Complex_base@N@std@@QAEAAV01@ABN@Z(ptr ptr) complex_double_mult_assign_double
+@ cdecl -arch=win64 ??X?$_Complex_base@N@std@@QEAAAEAV01@AEBN@Z(ptr ptr) complex_double_mult_assign_double
+@ cdecl -arch=arm ??X?$_Complex_base@O@std@@QAAAAV01@ABO@Z(ptr ptr) complex_double_mult_assign_double
+@ thiscall -arch=i386 ??X?$_Complex_base@O@std@@QAEAAV01@ABO@Z(ptr ptr) complex_double_mult_assign_double
+@ cdecl -arch=win64 ??X?$_Complex_base@O@std@@QEAAAEAV01@AEBO@Z(ptr ptr) complex_double_mult_assign_double
+@ cdecl -arch=win32 ??Xstd@@YAAAV?$complex@M@0@AAV10@ABV10@@Z(ptr ptr) complex_float_mult_assign_tmpl
+@ cdecl -arch=win64 ??Xstd@@YAAEAV?$complex@M@0@AEAV10@AEBV10@@Z(ptr ptr) complex_float_mult_assign_tmpl
+@ cdecl -arch=win32 ??Xstd@@YAAAV?$complex@N@0@AAV10@ABV10@@Z(ptr ptr) complex_double_mult_assign_tmpl
+@ cdecl -arch=win64 ??Xstd@@YAAEAV?$complex@N@0@AEAV10@AEBV10@@Z(ptr ptr) complex_double_mult_assign_tmpl
+@ cdecl -arch=win32 ??Xstd@@YAAAV?$complex@O@0@AAV10@ABV10@@Z(ptr ptr) complex_double_mult_assign_tmpl
+@ cdecl -arch=win64 ??Xstd@@YAAEAV?$complex@O@0@AEAV10@AEBV10@@Z(ptr ptr) complex_double_mult_assign_tmpl
+@ cdecl -arch=arm ??Y?$_Complex_base@M@std@@QAAAAV01@ABM@Z(ptr ptr) complex_float_add_assign_float
+@ thiscall -arch=i386 ??Y?$_Complex_base@M@std@@QAEAAV01@ABM@Z(ptr ptr) complex_float_add_assign_float
+@ cdecl -arch=win64 ??Y?$_Complex_base@M@std@@QEAAAEAV01@AEBM@Z(ptr ptr) complex_float_add_assign_float
+@ cdecl -arch=arm ??Y?$_Complex_base@N@std@@QAAAAV01@ABN@Z(ptr ptr) complex_double_add_assign_double
+@ thiscall -arch=i386 ??Y?$_Complex_base@N@std@@QAEAAV01@ABN@Z(ptr ptr) complex_double_add_assign_double
+@ cdecl -arch=win64 ??Y?$_Complex_base@N@std@@QEAAAEAV01@AEBN@Z(ptr ptr) complex_double_add_assign_double
+@ cdecl -arch=arm ??Y?$_Complex_base@O@std@@QAAAAV01@ABO@Z(ptr ptr) complex_double_add_assign_double
+@ thiscall -arch=i386 ??Y?$_Complex_base@O@std@@QAEAAV01@ABO@Z(ptr ptr) complex_double_add_assign_double
+@ cdecl -arch=win64 ??Y?$_Complex_base@O@std@@QEAAAEAV01@AEBO@Z(ptr ptr) complex_double_add_assign_double
 @ cdecl -arch=arm ??Y?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@QAAAAV01@ABV01@@Z(ptr ptr) MSVCP_basic_string_char_append
 @ thiscall -arch=i386 ??Y?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@QAEAAV01@ABV01@@Z(ptr ptr) MSVCP_basic_string_char_append
 @ cdecl -arch=win64 ??Y?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@QEAAAEAV01@AEBV01@@Z(ptr ptr) MSVCP_basic_string_char_append
@@ -2199,42 +2199,42 @@
 @ cdecl -arch=arm ??Y?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@QAAAAV01@PBG@Z(ptr wstr) MSVCP_basic_string_wchar_append_cstr
 @ thiscall -arch=i386 ??Y?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@QAEAAV01@PBG@Z(ptr wstr) MSVCP_basic_string_wchar_append_cstr
 @ cdecl -arch=win64 ??Y?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@QEAAAEAV01@PEBG@Z(ptr wstr) MSVCP_basic_string_wchar_append_cstr
-@ stub -arch=win32 ??Ystd@@YAAAV?$complex@M@0@AAV10@ABV10@@Z
-@ stub -arch=win64 ??Ystd@@YAAEAV?$complex@M@0@AEAV10@AEBV10@@Z
-@ stub -arch=win32 ??Ystd@@YAAAV?$complex@N@0@AAV10@ABV10@@Z
-@ stub -arch=win64 ??Ystd@@YAAEAV?$complex@N@0@AEAV10@AEBV10@@Z
-@ stub -arch=win32 ??Ystd@@YAAAV?$complex@O@0@AAV10@ABV10@@Z
-@ stub -arch=win64 ??Ystd@@YAAEAV?$complex@O@0@AEAV10@AEBV10@@Z
-@ stub -arch=arm ??Z?$_Complex_base@M@std@@QAAAAV01@ABM@Z
-@ stub -arch=i386 ??Z?$_Complex_base@M@std@@QAEAAV01@ABM@Z
-@ stub -arch=win64 ??Z?$_Complex_base@M@std@@QEAAAEAV01@AEBM@Z
-@ stub -arch=arm ??Z?$_Complex_base@N@std@@QAAAAV01@ABN@Z
-@ stub -arch=i386 ??Z?$_Complex_base@N@std@@QAEAAV01@ABN@Z
-@ stub -arch=win64 ??Z?$_Complex_base@N@std@@QEAAAEAV01@AEBN@Z
-@ stub -arch=arm ??Z?$_Complex_base@O@std@@QAAAAV01@ABO@Z
-@ stub -arch=i386 ??Z?$_Complex_base@O@std@@QAEAAV01@ABO@Z
-@ stub -arch=win64 ??Z?$_Complex_base@O@std@@QEAAAEAV01@AEBO@Z
-@ stub -arch=win32 ??Zstd@@YAAAV?$complex@M@0@AAV10@ABV10@@Z
-@ stub -arch=win64 ??Zstd@@YAAEAV?$complex@M@0@AEAV10@AEBV10@@Z
-@ stub -arch=win32 ??Zstd@@YAAAV?$complex@N@0@AAV10@ABV10@@Z
-@ stub -arch=win64 ??Zstd@@YAAEAV?$complex@N@0@AEAV10@AEBV10@@Z
-@ stub -arch=win32 ??Zstd@@YAAAV?$complex@O@0@AAV10@ABV10@@Z
-@ stub -arch=win64 ??Zstd@@YAAEAV?$complex@O@0@AEAV10@AEBV10@@Z
-@ stub -arch=arm ??_0?$_Complex_base@M@std@@QAAAAV01@ABM@Z
-@ stub -arch=i386 ??_0?$_Complex_base@M@std@@QAEAAV01@ABM@Z
-@ stub -arch=win64 ??_0?$_Complex_base@M@std@@QEAAAEAV01@AEBM@Z
-@ stub -arch=arm ??_0?$_Complex_base@N@std@@QAAAAV01@ABN@Z
-@ stub -arch=i386 ??_0?$_Complex_base@N@std@@QAEAAV01@ABN@Z
-@ stub -arch=win64 ??_0?$_Complex_base@N@std@@QEAAAEAV01@AEBN@Z
-@ stub -arch=arm ??_0?$_Complex_base@O@std@@QAAAAV01@ABO@Z
-@ stub -arch=i386 ??_0?$_Complex_base@O@std@@QAEAAV01@ABO@Z
-@ stub -arch=win64 ??_0?$_Complex_base@O@std@@QEAAAEAV01@AEBO@Z
-@ stub -arch=win32 ??_0std@@YAAAV?$complex@M@0@AAV10@ABV10@@Z
-@ stub -arch=win64 ??_0std@@YAAEAV?$complex@M@0@AEAV10@AEBV10@@Z
-@ stub -arch=win32 ??_0std@@YAAAV?$complex@N@0@AAV10@ABV10@@Z
-@ stub -arch=win64 ??_0std@@YAAEAV?$complex@N@0@AEAV10@AEBV10@@Z
-@ stub -arch=win32 ??_0std@@YAAAV?$complex@O@0@AAV10@ABV10@@Z
-@ stub -arch=win64 ??_0std@@YAAEAV?$complex@O@0@AEAV10@AEBV10@@Z
+@ cdecl -arch=win32 ??Ystd@@YAAAV?$complex@M@0@AAV10@ABV10@@Z(ptr ptr) complex_float_add_assign_tmpl
+@ cdecl -arch=win64 ??Ystd@@YAAEAV?$complex@M@0@AEAV10@AEBV10@@Z(ptr ptr) complex_float_add_assign_tmpl
+@ cdecl -arch=win32 ??Ystd@@YAAAV?$complex@N@0@AAV10@ABV10@@Z(ptr ptr) complex_double_add_assign_tmpl
+@ cdecl -arch=win64 ??Ystd@@YAAEAV?$complex@N@0@AEAV10@AEBV10@@Z(ptr ptr) complex_double_add_assign_tmpl
+@ cdecl -arch=win32 ??Ystd@@YAAAV?$complex@O@0@AAV10@ABV10@@Z(ptr ptr) complex_double_add_assign_tmpl
+@ cdecl -arch=win64 ??Ystd@@YAAEAV?$complex@O@0@AEAV10@AEBV10@@Z(ptr ptr) complex_double_add_assign_tmpl
+@ cdecl -arch=arm ??Z?$_Complex_base@M@std@@QAAAAV01@ABM@Z(ptr ptr) complex_float_sub_assign_float
+@ thiscall -arch=i386 ??Z?$_Complex_base@M@std@@QAEAAV01@ABM@Z(ptr ptr) complex_float_sub_assign_float
+@ cdecl -arch=win64 ??Z?$_Complex_base@M@std@@QEAAAEAV01@AEBM@Z(ptr ptr) complex_float_sub_assign_float
+@ cdecl -arch=arm ??Z?$_Complex_base@N@std@@QAAAAV01@ABN@Z(ptr ptr) complex_double_sub_assign_double
+@ thiscall -arch=i386 ??Z?$_Complex_base@N@std@@QAEAAV01@ABN@Z(ptr ptr) complex_double_sub_assign_double
+@ cdecl -arch=win64 ??Z?$_Complex_base@N@std@@QEAAAEAV01@AEBN@Z(ptr ptr) complex_double_sub_assign_double
+@ cdecl -arch=arm ??Z?$_Complex_base@O@std@@QAAAAV01@ABO@Z(ptr ptr) complex_double_sub_assign_double
+@ thiscall -arch=i386 ??Z?$_Complex_base@O@std@@QAEAAV01@ABO@Z(ptr ptr) complex_double_sub_assign_double
+@ cdecl -arch=win64 ??Z?$_Complex_base@O@std@@QEAAAEAV01@AEBO@Z(ptr ptr) complex_double_sub_assign_double
+@ cdecl -arch=win32 ??Zstd@@YAAAV?$complex@M@0@AAV10@ABV10@@Z(ptr ptr) complex_float_sub_assign_tmpl
+@ cdecl -arch=win64 ??Zstd@@YAAEAV?$complex@M@0@AEAV10@AEBV10@@Z(ptr ptr) complex_float_sub_assign_tmpl
+@ cdecl -arch=win32 ??Zstd@@YAAAV?$complex@N@0@AAV10@ABV10@@Z(ptr ptr) complex_double_sub_assign_tmpl
+@ cdecl -arch=win64 ??Zstd@@YAAEAV?$complex@N@0@AEAV10@AEBV10@@Z(ptr ptr) complex_double_sub_assign_tmpl
+@ cdecl -arch=win32 ??Zstd@@YAAAV?$complex@O@0@AAV10@ABV10@@Z(ptr ptr) complex_double_sub_assign_tmpl
+@ cdecl -arch=win64 ??Zstd@@YAAEAV?$complex@O@0@AEAV10@AEBV10@@Z(ptr ptr) complex_double_sub_assign_tmpl
+@ cdecl -arch=arm ??_0?$_Complex_base@M@std@@QAAAAV01@ABM@Z(ptr ptr) complex_float_div_assign_float
+@ thiscall -arch=i386 ??_0?$_Complex_base@M@std@@QAEAAV01@ABM@Z(ptr ptr) complex_float_div_assign_float
+@ cdecl -arch=win64 ??_0?$_Complex_base@M@std@@QEAAAEAV01@AEBM@Z(ptr ptr) complex_float_div_assign_float
+@ cdecl -arch=arm ??_0?$_Complex_base@N@std@@QAAAAV01@ABN@Z(ptr ptr) complex_double_div_assign_double
+@ thiscall -arch=i386 ??_0?$_Complex_base@N@std@@QAEAAV01@ABN@Z(ptr ptr) complex_double_div_assign_double
+@ cdecl -arch=win64 ??_0?$_Complex_base@N@std@@QEAAAEAV01@AEBN@Z(ptr ptr) complex_double_div_assign_double
+@ cdecl -arch=arm ??_0?$_Complex_base@O@std@@QAAAAV01@ABO@Z(ptr ptr) complex_double_div_assign_double
+@ thiscall -arch=i386 ??_0?$_Complex_base@O@std@@QAEAAV01@ABO@Z(ptr ptr) complex_double_div_assign_double
+@ cdecl -arch=win64 ??_0?$_Complex_base@O@std@@QEAAAEAV01@AEBO@Z(ptr ptr) complex_double_div_assign_double
+@ cdecl -arch=win32 ??_0std@@YAAAV?$complex@M@0@AAV10@ABV10@@Z(ptr ptr) complex_float_div_assign_tmpl
+@ cdecl -arch=win64 ??_0std@@YAAEAV?$complex@M@0@AEAV10@AEBV10@@Z(ptr ptr) complex_float_div_assign_tmpl
+@ cdecl -arch=win32 ??_0std@@YAAAV?$complex@N@0@AAV10@ABV10@@Z(ptr ptr) complex_double_div_assign_tmpl
+@ cdecl -arch=win64 ??_0std@@YAAEAV?$complex@N@0@AEAV10@AEBV10@@Z(ptr ptr) complex_double_div_assign_tmpl
+@ cdecl -arch=win32 ??_0std@@YAAAV?$complex@O@0@AAV10@ABV10@@Z(ptr ptr) complex_double_div_assign_tmpl
+@ cdecl -arch=win64 ??_0std@@YAAEAV?$complex@O@0@AEAV10@AEBV10@@Z(ptr ptr) complex_double_div_assign_tmpl
 @ extern ??_7?$_Mpunct@D@std@@6B@ mpunct_char_vtable
 @ extern ??_7?$_Mpunct@G@std@@6B@ mpunct_short_vtable
 @ extern ??_7?$basic_filebuf@DU?$char_traits@D@std@@@std@@6B@ basic_filebuf_char_vtable
@@ -2574,9 +2574,9 @@
 @ cdecl -arch=arm ?_Copy@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AAAXI@Z(ptr long) basic_string_wchar__Copy
 @ thiscall -arch=i386 ?_Copy@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AAEXI@Z(ptr long) basic_string_wchar__Copy
 @ cdecl -arch=win64 ?_Copy@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AEAAX_K@Z(ptr long) basic_string_wchar__Copy
-@ stub ?_Cosh@?$_Ctr@M@std@@SAMMM@Z
-@ stub ?_Cosh@?$_Ctr@N@std@@SANNN@Z
-@ stub ?_Cosh@?$_Ctr@O@std@@SAOOO@Z
+@ cdecl ?_Cosh@?$_Ctr@M@std@@SAMMM@Z(float float) _FCosh
+@ cdecl ?_Cosh@?$_Ctr@N@std@@SANNN@Z(double double) _Cosh
+@ cdecl ?_Cosh@?$_Ctr@O@std@@SAOOO@Z(double double) _Cosh
 @ cdecl -arch=arm ?_Decref@facet@locale@std@@QAAPAV123@XZ(ptr) locale_facet__Decref
 @ thiscall -arch=i386 ?_Decref@facet@locale@std@@QAEPAV123@XZ(ptr) locale_facet__Decref
 @ cdecl -arch=win64 ?_Decref@facet@locale@std@@QEAAPEAV123@XZ(ptr) locale_facet__Decref
@@ -2622,18 +2622,18 @@
 @ cdecl -arch=arm ?_Eos@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AAAXI@Z(ptr long) basic_string_wchar__Eos
 @ thiscall -arch=i386 ?_Eos@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AAEXI@Z(ptr long) basic_string_wchar__Eos
 @ cdecl -arch=win64 ?_Eos@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AEAAX_K@Z(ptr long) basic_string_wchar__Eos
-@ stub -arch=win32 ?_Exp@?$_Ctr@M@std@@SAFPAMMF@Z
-@ stub -arch=win64 ?_Exp@?$_Ctr@M@std@@SAFPEAMMF@Z
-@ stub -arch=win32 ?_Exp@?$_Ctr@N@std@@SAFPANNF@Z
-@ stub -arch=win64 ?_Exp@?$_Ctr@N@std@@SAFPEANNF@Z
-@ stub -arch=win32 ?_Exp@?$_Ctr@O@std@@SAFPAOOF@Z
-@ stub -arch=win64 ?_Exp@?$_Ctr@O@std@@SAFPEAOOF@Z
-@ stub -arch=win32 ?_Fabs@std@@YAMABV?$complex@M@1@PAH@Z
-@ stub -arch=win64 ?_Fabs@std@@YAMAEBV?$complex@M@1@PEAH@Z
-@ stub -arch=win32 ?_Fabs@std@@YANABV?$complex@N@1@PAH@Z
-@ stub -arch=win64 ?_Fabs@std@@YANAEBV?$complex@N@1@PEAH@Z
-@ stub -arch=win32 ?_Fabs@std@@YAOABV?$complex@O@1@PAH@Z
-@ stub -arch=win64 ?_Fabs@std@@YAOAEBV?$complex@O@1@PEAH@Z
+@ cdecl -arch=win32 ?_Exp@?$_Ctr@M@std@@SAFPAMMF@Z(ptr float long) _FExp
+@ cdecl -arch=win64 ?_Exp@?$_Ctr@M@std@@SAFPEAMMF@Z(ptr float long) _FExp
+@ cdecl -arch=win32 ?_Exp@?$_Ctr@N@std@@SAFPANNF@Z(ptr double long) _Exp
+@ cdecl -arch=win64 ?_Exp@?$_Ctr@N@std@@SAFPEANNF@Z(ptr double long) _Exp
+@ cdecl -arch=win32 ?_Exp@?$_Ctr@O@std@@SAFPAOOF@Z(ptr double long) _Exp
+@ cdecl -arch=win64 ?_Exp@?$_Ctr@O@std@@SAFPEAOOF@Z(ptr double long) _Exp
+@ cdecl -arch=win32 ?_Fabs@std@@YAMABV?$complex@M@1@PAH@Z(ptr ptr) complex_float__Fabs
+@ cdecl -arch=win64 ?_Fabs@std@@YAMAEBV?$complex@M@1@PEAH@Z(ptr ptr) complex_float__Fabs
+@ cdecl -arch=win32 ?_Fabs@std@@YANABV?$complex@N@1@PAH@Z(ptr ptr) complex_double__Fabs
+@ cdecl -arch=win64 ?_Fabs@std@@YANAEBV?$complex@N@1@PEAH@Z(ptr ptr) complex_double__Fabs
+@ cdecl -arch=win32 ?_Fabs@std@@YAOABV?$complex@O@1@PAH@Z(ptr ptr) complex_double__Fabs
+@ cdecl -arch=win64 ?_Fabs@std@@YAOAEBV?$complex@O@1@PEAH@Z(ptr ptr) complex_double__Fabs
 @ stub -arch=win32 ?_Ffmt@?$num_put@DV?$ostreambuf_iterator@DU?$char_traits@D@std@@@std@@@std@@KAPADPADDH@Z
 @ stub -arch=win64 ?_Ffmt@?$num_put@DV?$ostreambuf_iterator@DU?$char_traits@D@std@@@std@@@std@@KAPEADPEADDH@Z
 @ stub -arch=win32 ?_Ffmt@?$num_put@GV?$ostreambuf_iterator@GU?$char_traits@G@std@@@std@@@std@@KAPADPADDH@Z
@@ -2801,9 +2801,9 @@
 @ thiscall -arch=i386 ?_Incref@facet@locale@std@@QAEXXZ(ptr) locale_facet__Incref
 @ cdecl -arch=win64 ?_Incref@facet@locale@std@@QEAAXXZ(ptr) locale_facet__Incref
 @ extern ?_Index@ios_base@std@@0HA ios_base_Index
-@ stub ?_Infv@?$_Ctr@M@std@@SAMM@Z
-@ stub ?_Infv@?$_Ctr@N@std@@SANN@Z
-@ stub ?_Infv@?$_Ctr@O@std@@SAOO@Z
+@ cdecl ?_Infv@?$_Ctr@M@std@@SAMM@Z(float) std_Ctraits_float__Infv
+@ cdecl ?_Infv@?$_Ctr@N@std@@SANN@Z(double) std_Ctraits_double__Infv
+@ cdecl ?_Infv@?$_Ctr@O@std@@SAOO@Z(double) std_Ctraits_double__Infv
 @ cdecl -arch=arm ?_Init@?$_Mpunct@D@std@@IAAXABV_Locinfo@2@@Z(ptr ptr long) mpunct_char__Init
 @ thiscall -arch=i386 ?_Init@?$_Mpunct@D@std@@IAEXABV_Locinfo@2@@Z(ptr ptr long) mpunct_char__Init
 @ cdecl -arch=win64 ?_Init@?$_Mpunct@D@std@@IEAAXAEBV_Locinfo@2@@Z(ptr ptr long) mpunct_char__Init
@@ -2923,12 +2923,12 @@
 @ cdecl -arch=arm ?_Iscloc@locale@std@@QBA_NXZ(ptr) locale__Iscloc
 @ thiscall -arch=i386 ?_Iscloc@locale@std@@QBE_NXZ(ptr) locale__Iscloc
 @ cdecl -arch=win64 ?_Iscloc@locale@std@@QEBA_NXZ(ptr) locale__Iscloc
-@ stub ?_Isinf@?$_Ctr@M@std@@SA_NM@Z
-@ stub ?_Isinf@?$_Ctr@N@std@@SA_NN@Z
-@ stub ?_Isinf@?$_Ctr@O@std@@SA_NO@Z
-@ stub ?_Isnan@?$_Ctr@M@std@@SA_NM@Z
-@ stub ?_Isnan@?$_Ctr@N@std@@SA_NN@Z
-@ stub ?_Isnan@?$_Ctr@O@std@@SA_NO@Z
+@ cdecl ?_Isinf@?$_Ctr@M@std@@SA_NM@Z(float) std_Ctraits_float__Isinf
+@ cdecl ?_Isinf@?$_Ctr@N@std@@SA_NN@Z(double) std_Ctraits_double__Isinf
+@ cdecl ?_Isinf@?$_Ctr@O@std@@SA_NO@Z(double) std_Ctraits_double__Isinf
+@ cdecl ?_Isnan@?$_Ctr@M@std@@SA_NM@Z(float) std_Ctraits_float__Isnan
+@ cdecl ?_Isnan@?$_Ctr@N@std@@SA_NN@Z(double) std_Ctraits_double__Isnan
+@ cdecl ?_Isnan@?$_Ctr@O@std@@SA_NO@Z(double) std_Ctraits_long_double__Isnan
 @ cdecl -arch=arm ?_Makpat@?$_Mpunct@D@std@@AAAXAAUpattern@money_base@2@DDD@Z(ptr ptr long long long) mpunct_char__Makpat
 @ thiscall -arch=i386 ?_Makpat@?$_Mpunct@D@std@@AAEXAAUpattern@money_base@2@DDD@Z(ptr ptr long long long) mpunct_char__Makpat
 @ cdecl -arch=win64 ?_Makpat@?$_Mpunct@D@std@@AEAAXAEAUpattern@money_base@2@DDD@Z(ptr ptr long long long) mpunct_char__Makpat
@@ -2941,9 +2941,9 @@
 @ cdecl -arch=arm ?_Mode@?$basic_stringbuf@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AAAHH@Z(ptr long) basic_stringbuf_short__Getstate
 @ thiscall -arch=i386 ?_Mode@?$basic_stringbuf@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AAEHH@Z(ptr long) basic_stringbuf_short__Getstate
 @ cdecl -arch=win64 ?_Mode@?$basic_stringbuf@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AEAAHH@Z(ptr long) basic_stringbuf_short__Getstate
-@ stub ?_Nanv@?$_Ctr@M@std@@SAMM@Z
-@ stub ?_Nanv@?$_Ctr@N@std@@SANN@Z
-@ stub ?_Nanv@?$_Ctr@O@std@@SAOO@Z
+@ cdecl ?_Nanv@?$_Ctr@M@std@@SAMM@Z(float) std_Ctraits_float__Nanv
+@ cdecl ?_Nanv@?$_Ctr@N@std@@SANN@Z(double) std_Ctraits_double__Nanv
+@ cdecl ?_Nanv@?$_Ctr@O@std@@SAOO@Z(double) std_Ctraits_double__Nanv
 @ cdecl ?_Nomemory@std@@YAXXZ() _Nomemory
 @ cdecl -arch=win32 ?_Nullstr@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@CAPBDXZ() basic_string_char__Nullstr
 @ cdecl -arch=win64 ?_Nullstr@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@CAPEBDXZ() basic_string_char__Nullstr
@@ -3003,9 +3003,9 @@
 @ stub -arch=win64 ?_Rep@?$num_put@DV?$ostreambuf_iterator@DU?$char_traits@D@std@@@std@@@std@@KA?AV?$ostreambuf_iterator@DU?$char_traits@D@std@@@2@V32@D_K@Z
 @ stub -arch=win32 ?_Rep@?$num_put@GV?$ostreambuf_iterator@GU?$char_traits@G@std@@@std@@@std@@KA?AV?$ostreambuf_iterator@GU?$char_traits@G@std@@@2@V32@GI@Z
 @ stub -arch=win64 ?_Rep@?$num_put@GV?$ostreambuf_iterator@GU?$char_traits@G@std@@@std@@@std@@KA?AV?$ostreambuf_iterator@GU?$char_traits@G@std@@@2@V32@G_K@Z
-@ stub ?_Sinh@?$_Ctr@M@std@@SAMMM@Z
-@ stub ?_Sinh@?$_Ctr@N@std@@SANNN@Z
-@ stub ?_Sinh@?$_Ctr@O@std@@SAOOO@Z
+@ cdecl ?_Sinh@?$_Ctr@M@std@@SAMMM@Z(float float) _FSinh
+@ cdecl ?_Sinh@?$_Ctr@N@std@@SANNN@Z(double double) _Sinh
+@ cdecl ?_Sinh@?$_Ctr@O@std@@SAOOO@Z(double double) _Sinh
 @ cdecl -arch=arm ?_Split@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AAAXXZ(ptr) basic_string_char__Split
 @ thiscall -arch=i386 ?_Split@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AAEXXZ(ptr) basic_string_char__Split
 @ cdecl -arch=win64 ?_Split@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AEAAXXZ(ptr) basic_string_char__Split
@@ -3051,12 +3051,12 @@
 @ cdecl ?_Xran@std@@YAXXZ() _Xran
 @ cdecl -arch=win32 ?__Fiopen@std@@YAPAU_iobuf@@PBDH@Z(str long) ___Fiopen
 @ cdecl -arch=win64 ?__Fiopen@std@@YAPEAU_iobuf@@PEBDH@Z(str long) ___Fiopen
-@ stub -arch=win32 ?abs@std@@YAMABV?$complex@M@1@@Z
-@ stub -arch=win64 ?abs@std@@YAMAEBV?$complex@M@1@@Z
-@ stub -arch=win32 ?abs@std@@YANABV?$complex@N@1@@Z
-@ stub -arch=win64 ?abs@std@@YANAEBV?$complex@N@1@@Z
-@ stub -arch=win32 ?abs@std@@YAOABV?$complex@O@1@@Z
-@ stub -arch=win64 ?abs@std@@YAOAEBV?$complex@O@1@@Z
+@ cdecl -arch=win32 ?abs@std@@YAMABV?$complex@M@1@@Z(ptr) complex_float_abs
+@ cdecl -arch=win64 ?abs@std@@YAMAEBV?$complex@M@1@@Z(ptr) complex_float_abs
+@ cdecl -arch=win32 ?abs@std@@YANABV?$complex@N@1@@Z(ptr) complex_double_abs
+@ cdecl -arch=win64 ?abs@std@@YANAEBV?$complex@N@1@@Z(ptr) complex_double_abs
+@ cdecl -arch=win32 ?abs@std@@YAOABV?$complex@O@1@@Z(ptr) complex_double_abs
+@ cdecl -arch=win64 ?abs@std@@YAOAEBV?$complex@O@1@@Z(ptr) complex_double_abs
 @ cdecl -arch=arm ?always_noconv@codecvt_base@std@@QBA_NXZ(ptr) codecvt_base_always_noconv
 @ thiscall -arch=i386 ?always_noconv@codecvt_base@std@@QBE_NXZ(ptr) codecvt_base_always_noconv
 @ cdecl -arch=win64 ?always_noconv@codecvt_base@std@@QEBA_NXZ(ptr) codecvt_base_always_noconv
@@ -3096,12 +3096,12 @@
 @ cdecl -arch=arm ?append@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@QAAAAV12@PBGI@Z(ptr ptr long) MSVCP_basic_string_wchar_append_cstr_len
 @ thiscall -arch=i386 ?append@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@QAEAAV12@PBGI@Z(ptr ptr long) MSVCP_basic_string_wchar_append_cstr_len
 @ cdecl -arch=win64 ?append@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@QEAAAEAV12@_KG@Z(ptr long long) MSVCP_basic_string_wchar_append_len_ch
-@ stub -arch=win32 ?arg@std@@YAMABV?$complex@M@1@@Z
-@ stub -arch=win64 ?arg@std@@YAMAEBV?$complex@M@1@@Z
-@ stub -arch=win32 ?arg@std@@YANABV?$complex@N@1@@Z
-@ stub -arch=win64 ?arg@std@@YANAEBV?$complex@N@1@@Z
-@ stub -arch=win32 ?arg@std@@YAOABV?$complex@O@1@@Z
-@ stub -arch=win64 ?arg@std@@YAOAEBV?$complex@O@1@@Z
+@ cdecl -arch=win32 ?arg@std@@YAMABV?$complex@M@1@@Z(ptr) complex_float_arg
+@ cdecl -arch=win64 ?arg@std@@YAMAEBV?$complex@M@1@@Z(ptr) complex_float_arg
+@ cdecl -arch=win32 ?arg@std@@YANABV?$complex@N@1@@Z(ptr) complex_double_arg
+@ cdecl -arch=win64 ?arg@std@@YANAEBV?$complex@N@1@@Z(ptr) complex_double_arg
+@ cdecl -arch=win32 ?arg@std@@YAOABV?$complex@O@1@@Z(ptr) complex_double_arg
+@ cdecl -arch=win64 ?arg@std@@YAOAEBV?$complex@O@1@@Z(ptr) complex_double_arg
 @ cdecl -arch=arm ?assign@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@QAAAAV12@ABV12@@Z(ptr ptr) MSVCP_basic_string_char_assign
 @ thiscall -arch=i386 ?assign@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@QAEAAV12@ABV12@@Z(ptr ptr) MSVCP_basic_string_char_assign
 @ cdecl -arch=win64 ?assign@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@QEAAAEAV12@AEBV12@@Z(ptr ptr) MSVCP_basic_string_char_assign
@@ -3158,9 +3158,9 @@
 @ cdecl -arch=arm ?at@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@QBAABGI@Z(ptr long) MSVCP_basic_string_wchar_at
 @ thiscall -arch=i386 ?at@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@QBEABGI@Z(ptr long) MSVCP_basic_string_wchar_at
 @ cdecl -arch=win64 ?at@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@QEBAAEBG_K@Z(ptr long) MSVCP_basic_string_wchar_at
-@ stub ?atan2@?$_Ctr@M@std@@SAMMM@Z
-@ stub ?atan2@?$_Ctr@N@std@@SANNN@Z
-@ stub ?atan2@?$_Ctr@O@std@@SAOOO@Z
+@ cdecl ?atan2@?$_Ctr@M@std@@SAMMM@Z(float float) std_Ctraits_float_atan2
+@ cdecl ?atan2@?$_Ctr@N@std@@SANNN@Z(double double) std_Ctraits_double_atan2
+@ cdecl ?atan2@?$_Ctr@O@std@@SAOOO@Z(double double) std_Ctraits_long_double_atan2
 @ cdecl -arch=arm ?bad@ios_base@std@@QBA_NXZ(ptr) ios_base_bad
 @ thiscall -arch=i386 ?bad@ios_base@std@@QBE_NXZ(ptr) ios_base_bad
 @ cdecl -arch=win64 ?bad@ios_base@std@@QEBA_NXZ(ptr) ios_base_bad
@@ -3289,12 +3289,12 @@
 @ cdecl -arch=arm ?compare@?$collate@G@std@@QBAHPBG000@Z(ptr ptr ptr ptr ptr) collate_wchar_compare
 @ thiscall -arch=i386 ?compare@?$collate@G@std@@QBEHPBG000@Z(ptr ptr ptr ptr ptr) collate_wchar_compare
 @ cdecl -arch=win64 ?compare@?$collate@G@std@@QEBAHPEBG000@Z(ptr ptr ptr ptr ptr) collate_wchar_compare
-@ stub -arch=win32 ?conj@std@@YA?AV?$complex@M@1@ABV21@@Z
-@ stub -arch=win64 ?conj@std@@YA?AV?$complex@M@1@AEBV21@@Z
-@ stub -arch=win32 ?conj@std@@YA?AV?$complex@N@1@ABV21@@Z
-@ stub -arch=win64 ?conj@std@@YA?AV?$complex@N@1@AEBV21@@Z
-@ stub -arch=win32 ?conj@std@@YA?AV?$complex@O@1@ABV21@@Z
-@ stub -arch=win64 ?conj@std@@YA?AV?$complex@O@1@AEBV21@@Z
+@ cdecl -arch=win32 ?conj@std@@YA?AV?$complex@M@1@ABV21@@Z(ptr ptr) complex_float_conj
+@ cdecl -arch=win64 ?conj@std@@YA?AV?$complex@M@1@AEBV21@@Z(ptr ptr) complex_float_conj
+@ cdecl -arch=win32 ?conj@std@@YA?AV?$complex@N@1@ABV21@@Z(ptr ptr) complex_double_conj
+@ cdecl -arch=win64 ?conj@std@@YA?AV?$complex@N@1@AEBV21@@Z(ptr ptr) complex_double_conj
+@ cdecl -arch=win32 ?conj@std@@YA?AV?$complex@O@1@ABV21@@Z(ptr ptr) complex_double_conj
+@ cdecl -arch=win64 ?conj@std@@YA?AV?$complex@O@1@AEBV21@@Z(ptr ptr) complex_double_conj
 @ cdecl -arch=arm ?copy@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@QBAIPADII@Z(ptr ptr long long) basic_string_char_copy
 @ thiscall -arch=i386 ?copy@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@QBEIPADII@Z(ptr ptr long long) basic_string_char_copy
 @ cdecl -arch=win64 ?copy@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@QEBA_KPEAD_K1@Z(ptr ptr long long) basic_string_char_copy
@@ -3314,21 +3314,21 @@
 @ cdecl -arch=arm ?copyfmt@ios_base@std@@QAAAAV12@ABV12@@Z(ptr ptr) ios_base_copyfmt
 @ thiscall -arch=i386 ?copyfmt@ios_base@std@@QAEAAV12@ABV12@@Z(ptr ptr) ios_base_copyfmt
 @ cdecl -arch=win64 ?copyfmt@ios_base@std@@QEAAAEAV12@AEBV12@@Z(ptr ptr) ios_base_copyfmt
-@ stub ?cos@?$_Ctr@M@std@@SAMM@Z
-@ stub ?cos@?$_Ctr@N@std@@SANN@Z
-@ stub ?cos@?$_Ctr@O@std@@SAOO@Z
-@ stub -arch=win32 ?cos@std@@YA?AV?$complex@M@1@ABV21@@Z
-@ stub -arch=win64 ?cos@std@@YA?AV?$complex@M@1@AEBV21@@Z
-@ stub -arch=win32 ?cos@std@@YA?AV?$complex@N@1@ABV21@@Z
-@ stub -arch=win64 ?cos@std@@YA?AV?$complex@N@1@AEBV21@@Z
-@ stub -arch=win32 ?cos@std@@YA?AV?$complex@O@1@ABV21@@Z
-@ stub -arch=win64 ?cos@std@@YA?AV?$complex@O@1@AEBV21@@Z
-@ stub -arch=win32 ?cosh@std@@YA?AV?$complex@M@1@ABV21@@Z
-@ stub -arch=win64 ?cosh@std@@YA?AV?$complex@M@1@AEBV21@@Z
-@ stub -arch=win32 ?cosh@std@@YA?AV?$complex@N@1@ABV21@@Z
-@ stub -arch=win64 ?cosh@std@@YA?AV?$complex@N@1@AEBV21@@Z
-@ stub -arch=win32 ?cosh@std@@YA?AV?$complex@O@1@ABV21@@Z
-@ stub -arch=win64 ?cosh@std@@YA?AV?$complex@O@1@AEBV21@@Z
+@ cdecl ?cos@?$_Ctr@M@std@@SAMM@Z(float) std_Ctraits_float_cos
+@ cdecl ?cos@?$_Ctr@N@std@@SANN@Z(double) std_Ctraits_double_cos
+@ cdecl ?cos@?$_Ctr@O@std@@SAOO@Z(double) std_Ctraits_long_double_cos
+@ cdecl -arch=win32 ?cos@std@@YA?AV?$complex@M@1@ABV21@@Z(ptr ptr) complex_float_cos
+@ cdecl -arch=win64 ?cos@std@@YA?AV?$complex@M@1@AEBV21@@Z(ptr ptr) complex_float_cos
+@ cdecl -arch=win32 ?cos@std@@YA?AV?$complex@N@1@ABV21@@Z(ptr ptr) complex_double_cos
+@ cdecl -arch=win64 ?cos@std@@YA?AV?$complex@N@1@AEBV21@@Z(ptr ptr) complex_double_cos
+@ cdecl -arch=win32 ?cos@std@@YA?AV?$complex@O@1@ABV21@@Z(ptr ptr) complex_double_cos
+@ cdecl -arch=win64 ?cos@std@@YA?AV?$complex@O@1@AEBV21@@Z(ptr ptr) complex_double_cos
+@ cdecl -arch=win32 ?cosh@std@@YA?AV?$complex@M@1@ABV21@@Z(ptr ptr) complex_float_cosh
+@ cdecl -arch=win64 ?cosh@std@@YA?AV?$complex@M@1@AEBV21@@Z(ptr ptr) complex_float_cosh
+@ cdecl -arch=win32 ?cosh@std@@YA?AV?$complex@N@1@ABV21@@Z(ptr ptr) complex_double_cosh
+@ cdecl -arch=win64 ?cosh@std@@YA?AV?$complex@N@1@AEBV21@@Z(ptr ptr) complex_double_cosh
+@ cdecl -arch=win32 ?cosh@std@@YA?AV?$complex@O@1@ABV21@@Z(ptr ptr) complex_double_cosh
+@ cdecl -arch=win64 ?cosh@std@@YA?AV?$complex@O@1@AEBV21@@Z(ptr ptr) complex_double_cosh
 @ extern ?cout@std@@3V?$basic_ostream@DU?$char_traits@D@std@@@1@A cout
 @ cdecl -arch=arm ?curr_symbol@?$_Mpunct@D@std@@QBA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@2@XZ(ptr ptr) mpunct_char_curr_symbol
 @ thiscall -arch=i386 ?curr_symbol@?$_Mpunct@D@std@@QBE?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@2@XZ(ptr ptr) mpunct_char_curr_symbol
@@ -3842,15 +3842,15 @@
 @ cdecl -arch=arm ?exceptions@ios_base@std@@QBAHXZ(ptr) ios_base_exceptions_get
 @ thiscall -arch=i386 ?exceptions@ios_base@std@@QBEHXZ(ptr) ios_base_exceptions_get
 @ cdecl -arch=win64 ?exceptions@ios_base@std@@QEBAHXZ(ptr) ios_base_exceptions_get
-@ stub ?exp@?$_Ctr@M@std@@SAMM@Z
-@ stub ?exp@?$_Ctr@N@std@@SANN@Z
-@ stub ?exp@?$_Ctr@O@std@@SAOO@Z
-@ stub -arch=win32 ?exp@std@@YA?AV?$complex@M@1@ABV21@@Z
-@ stub -arch=win64 ?exp@std@@YA?AV?$complex@M@1@AEBV21@@Z
-@ stub -arch=win32 ?exp@std@@YA?AV?$complex@N@1@ABV21@@Z
-@ stub -arch=win64 ?exp@std@@YA?AV?$complex@N@1@AEBV21@@Z
-@ stub -arch=win32 ?exp@std@@YA?AV?$complex@O@1@ABV21@@Z
-@ stub -arch=win64 ?exp@std@@YA?AV?$complex@O@1@AEBV21@@Z
+@ cdecl ?exp@?$_Ctr@M@std@@SAMM@Z(float) std_Ctraits_float_exp
+@ cdecl ?exp@?$_Ctr@N@std@@SANN@Z(double) std_Ctraits_double_exp
+@ cdecl ?exp@?$_Ctr@O@std@@SAOO@Z(double) std_Ctraits_long_double_exp
+@ cdecl -arch=win32 ?exp@std@@YA?AV?$complex@M@1@ABV21@@Z(ptr ptr) complex_float_exp
+@ cdecl -arch=win64 ?exp@std@@YA?AV?$complex@M@1@AEBV21@@Z(ptr ptr) complex_float_exp
+@ cdecl -arch=win32 ?exp@std@@YA?AV?$complex@N@1@ABV21@@Z(ptr ptr) complex_double_exp
+@ cdecl -arch=win64 ?exp@std@@YA?AV?$complex@N@1@AEBV21@@Z(ptr ptr) complex_double_exp
+@ cdecl -arch=win32 ?exp@std@@YA?AV?$complex@O@1@ABV21@@Z(ptr ptr) complex_double_exp
+@ cdecl -arch=win64 ?exp@std@@YA?AV?$complex@O@1@AEBV21@@Z(ptr ptr) complex_double_exp
 @ cdecl -arch=arm ?fail@ios_base@std@@QBA_NXZ(ptr) ios_base_fail
 @ thiscall -arch=i386 ?fail@ios_base@std@@QBE_NXZ(ptr) ios_base_fail
 @ cdecl -arch=win64 ?fail@ios_base@std@@QEBA_NXZ(ptr) ios_base_fail
@@ -4272,29 +4272,29 @@
 @ thiscall -arch=i386 ?ignore@?$basic_istream@GU?$char_traits@G@std@@@std@@QAEAAV12@HG@Z(ptr long long) basic_istream_short_ignore
 @ cdecl -arch=win64 ?ignore@?$basic_istream@GU?$char_traits@G@std@@@std@@QEAAAEAV12@_JG@Z(ptr long long) basic_istream_short_ignore
 @ stub -arch=arm ?imag@?$_Complex_base@M@std@@QAAMABM@Z
-@ stub -arch=i386 ?imag@?$_Complex_base@M@std@@QAEMABM@Z
-@ stub -arch=win64 ?imag@?$_Complex_base@M@std@@QEAAMAEBM@Z
+@ thiscall -arch=i386 ?imag@?$_Complex_base@M@std@@QAEMABM@Z(ptr ptr) complex_float_imag_set
+@ cdecl -arch=win64 ?imag@?$_Complex_base@M@std@@QEAAMAEBM@Z(ptr ptr) complex_float_imag_set
 @ stub -arch=arm ?imag@?$_Complex_base@M@std@@QBAMXZ
-@ stub -arch=i386 ?imag@?$_Complex_base@M@std@@QBEMXZ
-@ stub -arch=win64 ?imag@?$_Complex_base@M@std@@QEBAMXZ
+@ thiscall -arch=i386 ?imag@?$_Complex_base@M@std@@QBEMXZ(ptr) complex_float_imag_get
+@ cdecl -arch=win64 ?imag@?$_Complex_base@M@std@@QEBAMXZ(ptr) complex_float_imag_get
 @ stub -arch=arm ?imag@?$_Complex_base@N@std@@QAANABN@Z
-@ stub -arch=i386 ?imag@?$_Complex_base@N@std@@QAENABN@Z
-@ stub -arch=win64 ?imag@?$_Complex_base@N@std@@QEAANAEBN@Z
+@ thiscall -arch=i386 ?imag@?$_Complex_base@N@std@@QAENABN@Z(ptr ptr) complex_double_imag_set
+@ cdecl -arch=win64 ?imag@?$_Complex_base@N@std@@QEAANAEBN@Z(ptr ptr) complex_double_imag_set
 @ stub -arch=arm ?imag@?$_Complex_base@N@std@@QBANXZ
-@ stub -arch=i386 ?imag@?$_Complex_base@N@std@@QBENXZ
-@ stub -arch=win64 ?imag@?$_Complex_base@N@std@@QEBANXZ
+@ thiscall -arch=i386 ?imag@?$_Complex_base@N@std@@QBENXZ(ptr) complex_double_imag_get
+@ cdecl -arch=win64 ?imag@?$_Complex_base@N@std@@QEBANXZ(ptr) complex_double_imag_get
 @ stub -arch=arm ?imag@?$_Complex_base@O@std@@QAAOABO@Z
-@ stub -arch=i386 ?imag@?$_Complex_base@O@std@@QAEOABO@Z
-@ stub -arch=win64 ?imag@?$_Complex_base@O@std@@QEAAOAEBO@Z
+@ thiscall -arch=i386 ?imag@?$_Complex_base@O@std@@QAEOABO@Z(ptr ptr) complex_double_imag_set
+@ cdecl -arch=win64 ?imag@?$_Complex_base@O@std@@QEAAOAEBO@Z(ptr ptr) complex_double_imag_set
 @ stub -arch=arm ?imag@?$_Complex_base@O@std@@QBAOXZ
-@ stub -arch=i386 ?imag@?$_Complex_base@O@std@@QBEOXZ
-@ stub -arch=win64 ?imag@?$_Complex_base@O@std@@QEBAOXZ
-@ stub -arch=win32 ?imag@std@@YAMABV?$complex@M@1@@Z
-@ stub -arch=win64 ?imag@std@@YAMAEBV?$complex@M@1@@Z
-@ stub -arch=win32 ?imag@std@@YANABV?$complex@N@1@@Z
-@ stub -arch=win64 ?imag@std@@YANAEBV?$complex@N@1@@Z
-@ stub -arch=win32 ?imag@std@@YAOABV?$complex@O@1@@Z
-@ stub -arch=win64 ?imag@std@@YAOAEBV?$complex@O@1@@Z
+@ thiscall -arch=i386 ?imag@?$_Complex_base@O@std@@QBEOXZ(ptr) complex_double_imag_get
+@ cdecl -arch=win64 ?imag@?$_Complex_base@O@std@@QEBAOXZ(ptr) complex_double_imag_get
+@ cdecl -arch=win32 ?imag@std@@YAMABV?$complex@M@1@@Z(ptr) complex_float_imag
+@ cdecl -arch=win64 ?imag@std@@YAMAEBV?$complex@M@1@@Z(ptr) complex_float_imag
+@ cdecl -arch=win32 ?imag@std@@YANABV?$complex@N@1@@Z(ptr) complex_double_imag
+@ cdecl -arch=win64 ?imag@std@@YANAEBV?$complex@N@1@@Z(ptr) complex_double_imag
+@ cdecl -arch=win32 ?imag@std@@YAOABV?$complex@O@1@@Z(ptr) complex_double_imag
+@ cdecl -arch=win64 ?imag@std@@YAOAEBV?$complex@O@1@@Z(ptr) complex_double_imag
 @ cdecl -arch=arm ?imbue@?$basic_ios@DU?$char_traits@D@std@@@std@@QAA?AVlocale@2@ABV32@@Z(ptr ptr ptr) basic_ios_char_imbue
 @ thiscall -arch=i386 ?imbue@?$basic_ios@DU?$char_traits@D@std@@@std@@QAE?AVlocale@2@ABV32@@Z(ptr ptr ptr) basic_ios_char_imbue
 @ cdecl -arch=win64 ?imbue@?$basic_ios@DU?$char_traits@D@std@@@std@@QEAA?AVlocale@2@AEBV32@@Z(ptr ptr ptr) basic_ios_char_imbue
@@ -4444,9 +4444,9 @@
 @ cdecl -arch=arm ?iword@ios_base@std@@QAAAAJH@Z(ptr long) ios_base_iword
 @ thiscall -arch=i386 ?iword@ios_base@std@@QAEAAJH@Z(ptr long) ios_base_iword
 @ cdecl -arch=win64 ?iword@ios_base@std@@QEAAAEAJH@Z(ptr long) ios_base_iword
-@ stub ?ldexp@?$_Ctr@M@std@@SAMMH@Z
-@ stub ?ldexp@?$_Ctr@N@std@@SANNH@Z
-@ stub ?ldexp@?$_Ctr@O@std@@SAOOH@Z
+@ cdecl ?ldexp@?$_Ctr@M@std@@SAMMH@Z(float long) std_Ctraits_float_ldexp
+@ cdecl ?ldexp@?$_Ctr@N@std@@SANNH@Z(double long) std_Ctraits_double_ldexp
+@ cdecl ?ldexp@?$_Ctr@O@std@@SAOOH@Z(double long) std_Ctraits_long_double_ldexp
 @ cdecl -arch=arm ?length@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@QBAIXZ(ptr) MSVCP_basic_string_char_length
 @ thiscall -arch=i386 ?length@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@QBEIXZ(ptr) MSVCP_basic_string_char_length
 @ cdecl -arch=win64 ?length@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@QEBA_KXZ(ptr) MSVCP_basic_string_char_length
@@ -4463,21 +4463,21 @@
 @ stub -arch=arm ?length@?$codecvt@GDH@std@@QBAHAAHPBG1I@Z
 @ stub -arch=i386 ?length@?$codecvt@GDH@std@@QBEHAAHPBG1I@Z
 @ stub -arch=win64 ?length@?$codecvt@GDH@std@@QEBAHAEAHPEBG1_K@Z
-@ stub -arch=win32 ?log10@std@@YA?AV?$complex@M@1@ABV21@@Z
-@ stub -arch=win64 ?log10@std@@YA?AV?$complex@M@1@AEBV21@@Z
-@ stub -arch=win32 ?log10@std@@YA?AV?$complex@N@1@ABV21@@Z
-@ stub -arch=win64 ?log10@std@@YA?AV?$complex@N@1@AEBV21@@Z
-@ stub -arch=win32 ?log10@std@@YA?AV?$complex@O@1@ABV21@@Z
-@ stub -arch=win64 ?log10@std@@YA?AV?$complex@O@1@AEBV21@@Z
-@ stub ?log@?$_Ctr@M@std@@SAMM@Z
-@ stub ?log@?$_Ctr@N@std@@SANN@Z
-@ stub ?log@?$_Ctr@O@std@@SAOO@Z
-@ stub -arch=win32 ?log@std@@YA?AV?$complex@M@1@ABV21@@Z
-@ stub -arch=win64 ?log@std@@YA?AV?$complex@M@1@AEBV21@@Z
-@ stub -arch=win32 ?log@std@@YA?AV?$complex@N@1@ABV21@@Z
-@ stub -arch=win64 ?log@std@@YA?AV?$complex@N@1@AEBV21@@Z
-@ stub -arch=win32 ?log@std@@YA?AV?$complex@O@1@ABV21@@Z
-@ stub -arch=win64 ?log@std@@YA?AV?$complex@O@1@AEBV21@@Z
+@ cdecl -arch=win32 ?log10@std@@YA?AV?$complex@M@1@ABV21@@Z(ptr ptr) complex_float_log10
+@ cdecl -arch=win64 ?log10@std@@YA?AV?$complex@M@1@AEBV21@@Z(ptr ptr) complex_float_log10
+@ cdecl -arch=win32 ?log10@std@@YA?AV?$complex@N@1@ABV21@@Z(ptr ptr) complex_double_log10
+@ cdecl -arch=win64 ?log10@std@@YA?AV?$complex@N@1@AEBV21@@Z(ptr ptr) complex_double_log10
+@ cdecl -arch=win32 ?log10@std@@YA?AV?$complex@O@1@ABV21@@Z(ptr ptr) complex_double_log10
+@ cdecl -arch=win64 ?log10@std@@YA?AV?$complex@O@1@AEBV21@@Z(ptr ptr) complex_double_log10
+@ cdecl ?log@?$_Ctr@M@std@@SAMM@Z(float) std_Ctraits_float_log
+@ cdecl ?log@?$_Ctr@N@std@@SANN@Z(double) std_Ctraits_double_log
+@ cdecl ?log@?$_Ctr@O@std@@SAOO@Z(double) std_Ctraits_long_double_log
+@ cdecl -arch=win32 ?log@std@@YA?AV?$complex@M@1@ABV21@@Z(ptr ptr) complex_float_log
+@ cdecl -arch=win64 ?log@std@@YA?AV?$complex@M@1@AEBV21@@Z(ptr ptr) complex_float_log
+@ cdecl -arch=win32 ?log@std@@YA?AV?$complex@N@1@ABV21@@Z(ptr ptr) complex_double_log
+@ cdecl -arch=win64 ?log@std@@YA?AV?$complex@N@1@AEBV21@@Z(ptr ptr) complex_double_log
+@ cdecl -arch=win32 ?log@std@@YA?AV?$complex@O@1@ABV21@@Z(ptr ptr) complex_double_log
+@ cdecl -arch=win64 ?log@std@@YA?AV?$complex@O@1@AEBV21@@Z(ptr ptr) complex_double_log
 @ cdecl -arch=win32 ?lt@?$char_traits@D@std@@SA_NABD0@Z(ptr ptr) msvcp90.?lt@?$char_traits@D@std@@SA_NABD0@Z
 @ cdecl -arch=win64 ?lt@?$char_traits@D@std@@SA_NAEBD0@Z(ptr ptr) msvcp90.?lt@?$char_traits@D@std@@SA_NAEBD0@Z
 @ cdecl -arch=win32 ?lt@?$char_traits@G@std@@SA_NABG0@Z(ptr ptr) msvcp90.?lt@?$char_traits@G@std@@SA_NABG0@Z
@@ -4554,12 +4554,12 @@
 @ cdecl -arch=arm ?negative_sign@?$_Mpunct@G@std@@QBA?AV?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@2@XZ(ptr ptr) mpunct_short_negative_sign
 @ thiscall -arch=i386 ?negative_sign@?$_Mpunct@G@std@@QBE?AV?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@2@XZ(ptr ptr) mpunct_short_negative_sign
 @ cdecl -arch=win64 ?negative_sign@?$_Mpunct@G@std@@QEBA?AV?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@2@XZ(ptr ptr) mpunct_short_negative_sign
-@ stub -arch=win32 ?norm@std@@YAMABV?$complex@M@1@@Z
-@ stub -arch=win64 ?norm@std@@YAMAEBV?$complex@M@1@@Z
-@ stub -arch=win32 ?norm@std@@YANABV?$complex@N@1@@Z
-@ stub -arch=win64 ?norm@std@@YANAEBV?$complex@N@1@@Z
-@ stub -arch=win32 ?norm@std@@YAOABV?$complex@O@1@@Z
-@ stub -arch=win64 ?norm@std@@YAOAEBV?$complex@O@1@@Z
+@ cdecl -arch=win32 ?norm@std@@YAMABV?$complex@M@1@@Z(ptr) complex_float_norm
+@ cdecl -arch=win64 ?norm@std@@YAMAEBV?$complex@M@1@@Z(ptr) complex_float_norm
+@ cdecl -arch=win32 ?norm@std@@YANABV?$complex@N@1@@Z(ptr) complex_double_norm
+@ cdecl -arch=win64 ?norm@std@@YANAEBV?$complex@N@1@@Z(ptr) complex_double_norm
+@ cdecl -arch=win32 ?norm@std@@YAOABV?$complex@O@1@@Z(ptr) complex_double_norm
+@ cdecl -arch=win64 ?norm@std@@YAOAEBV?$complex@O@1@@Z(ptr) complex_double_norm
 @ cdecl -arch=win32 ?not_eof@?$char_traits@D@std@@SAHABH@Z(ptr) msvcp90.?not_eof@?$char_traits@D@std@@SAHABH@Z
 @ cdecl -arch=win64 ?not_eof@?$char_traits@D@std@@SAHAEBH@Z(ptr) msvcp90.?not_eof@?$char_traits@D@std@@SAHAEBH@Z
 @ cdecl -arch=win32 ?not_eof@?$char_traits@G@std@@SAGABG@Z(ptr) msvcp90.?not_eof@?$char_traits@G@std@@SAGABG@Z
@@ -4701,18 +4701,18 @@
 @ cdecl -arch=arm ?peek@?$basic_istream@GU?$char_traits@G@std@@@std@@QAAGXZ(ptr) basic_istream_short_peek
 @ thiscall -arch=i386 ?peek@?$basic_istream@GU?$char_traits@G@std@@@std@@QAEGXZ(ptr) basic_istream_short_peek
 @ cdecl -arch=win64 ?peek@?$basic_istream@GU?$char_traits@G@std@@@std@@QEAAGXZ(ptr) basic_istream_short_peek
-@ stub -arch=win32 ?polar@std@@YA?AV?$complex@M@1@ABM0@Z
-@ stub -arch=win64 ?polar@std@@YA?AV?$complex@M@1@AEBM0@Z
-@ stub -arch=win32 ?polar@std@@YA?AV?$complex@M@1@ABM@Z
-@ stub -arch=win64 ?polar@std@@YA?AV?$complex@M@1@AEBM@Z
-@ stub -arch=win32 ?polar@std@@YA?AV?$complex@N@1@ABN0@Z
-@ stub -arch=win64 ?polar@std@@YA?AV?$complex@N@1@AEBN0@Z
-@ stub -arch=win32 ?polar@std@@YA?AV?$complex@N@1@ABN@Z
-@ stub -arch=win64 ?polar@std@@YA?AV?$complex@N@1@AEBN@Z
-@ stub -arch=win32 ?polar@std@@YA?AV?$complex@O@1@ABO0@Z
-@ stub -arch=win64 ?polar@std@@YA?AV?$complex@O@1@AEBO0@Z
-@ stub -arch=win32 ?polar@std@@YA?AV?$complex@O@1@ABO@Z
-@ stub -arch=win64 ?polar@std@@YA?AV?$complex@O@1@AEBO@Z
+@ cdecl -arch=win32 ?polar@std@@YA?AV?$complex@M@1@ABM0@Z(ptr ptr ptr) complex_float_polar_theta
+@ cdecl -arch=win64 ?polar@std@@YA?AV?$complex@M@1@AEBM0@Z(ptr ptr ptr) complex_float_polar_theta
+@ cdecl -arch=win32 ?polar@std@@YA?AV?$complex@M@1@ABM@Z(ptr ptr) complex_float_polar
+@ cdecl -arch=win64 ?polar@std@@YA?AV?$complex@M@1@AEBM@Z(ptr ptr) complex_float_polar
+@ cdecl -arch=win32 ?polar@std@@YA?AV?$complex@N@1@ABN0@Z(ptr ptr ptr) complex_double_polar_theta
+@ cdecl -arch=win64 ?polar@std@@YA?AV?$complex@N@1@AEBN0@Z(ptr ptr ptr) complex_double_polar_theta
+@ cdecl -arch=win32 ?polar@std@@YA?AV?$complex@N@1@ABN@Z(ptr ptr) complex_double_polar
+@ cdecl -arch=win64 ?polar@std@@YA?AV?$complex@N@1@AEBN@Z(ptr ptr) complex_double_polar
+@ cdecl -arch=win32 ?polar@std@@YA?AV?$complex@O@1@ABO0@Z(ptr ptr ptr) complex_double_polar_theta
+@ cdecl -arch=win64 ?polar@std@@YA?AV?$complex@O@1@AEBO0@Z(ptr ptr ptr) complex_double_polar_theta
+@ cdecl -arch=win32 ?polar@std@@YA?AV?$complex@O@1@ABO@Z(ptr ptr) complex_double_polar
+@ cdecl -arch=win64 ?polar@std@@YA?AV?$complex@O@1@AEBO@Z(ptr ptr) complex_double_polar
 @ cdecl -arch=arm ?pos_format@?$_Mpunct@D@std@@QBA?AUpattern@money_base@2@XZ(ptr ptr) mpunct_char_pos_format
 @ thiscall -arch=i386 ?pos_format@?$_Mpunct@D@std@@QBE?AUpattern@money_base@2@XZ(ptr ptr) mpunct_char_pos_format
 @ cdecl -arch=win64 ?pos_format@?$_Mpunct@D@std@@QEBA?AUpattern@money_base@2@XZ(ptr ptr) mpunct_char_pos_format
@@ -4725,31 +4725,31 @@
 @ cdecl -arch=arm ?positive_sign@?$_Mpunct@G@std@@QBA?AV?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@2@XZ(ptr ptr) mpunct_short_positive_sign
 @ thiscall -arch=i386 ?positive_sign@?$_Mpunct@G@std@@QBE?AV?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@2@XZ(ptr ptr) mpunct_short_positive_sign
 @ cdecl -arch=win64 ?positive_sign@?$_Mpunct@G@std@@QEBA?AV?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@2@XZ(ptr ptr) mpunct_short_positive_sign
-@ stub ?pow@?$_Ctr@M@std@@SAMMM@Z
-@ stub ?pow@?$_Ctr@N@std@@SANNN@Z
-@ stub ?pow@?$_Ctr@O@std@@SAOOO@Z
-@ stub -arch=win32 ?pow@std@@YA?AV?$complex@M@1@ABMABV21@@Z
-@ stub -arch=win64 ?pow@std@@YA?AV?$complex@M@1@AEBMAEBV21@@Z
-@ stub -arch=win32 ?pow@std@@YA?AV?$complex@M@1@ABV21@0@Z
-@ stub -arch=win64 ?pow@std@@YA?AV?$complex@M@1@AEBV21@0@Z
-@ stub -arch=win32 ?pow@std@@YA?AV?$complex@M@1@ABV21@ABM@Z
-@ stub -arch=win64 ?pow@std@@YA?AV?$complex@M@1@AEBV21@AEBM@Z
+@ cdecl ?pow@?$_Ctr@M@std@@SAMMM@Z(float float) std_Ctraits_float_pow
+@ cdecl ?pow@?$_Ctr@N@std@@SANNN@Z(double double) std_Ctraits_double_pow
+@ cdecl ?pow@?$_Ctr@O@std@@SAOOO@Z(double double) std_Ctraits_long_double_pow
+@ cdecl -arch=win32 ?pow@std@@YA?AV?$complex@M@1@ABMABV21@@Z(ptr ptr ptr) complex_float_pow_fc
+@ cdecl -arch=win64 ?pow@std@@YA?AV?$complex@M@1@AEBMAEBV21@@Z(ptr ptr ptr) complex_float_pow_fc
+@ cdecl -arch=win32 ?pow@std@@YA?AV?$complex@M@1@ABV21@0@Z(ptr ptr ptr) complex_float_pow
+@ cdecl -arch=win64 ?pow@std@@YA?AV?$complex@M@1@AEBV21@0@Z(ptr ptr ptr) complex_float_pow
+@ cdecl -arch=win32 ?pow@std@@YA?AV?$complex@M@1@ABV21@ABM@Z(ptr ptr ptr) complex_float_pow_cf
+@ cdecl -arch=win64 ?pow@std@@YA?AV?$complex@M@1@AEBV21@AEBM@Z(ptr ptr ptr) complex_float_pow_cf
 @ cdecl -arch=win32 ?pow@std@@YA?AV?$complex@M@1@ABV21@H@Z(ptr ptr long) complex_float_pow_ci
 @ cdecl -arch=win64 ?pow@std@@YA?AV?$complex@M@1@AEBV21@H@Z(ptr ptr long) complex_float_pow_ci
-@ stub -arch=win32 ?pow@std@@YA?AV?$complex@N@1@ABNABV21@@Z
-@ stub -arch=win64 ?pow@std@@YA?AV?$complex@N@1@AEBNAEBV21@@Z
-@ stub -arch=win32 ?pow@std@@YA?AV?$complex@N@1@ABV21@0@Z
-@ stub -arch=win64 ?pow@std@@YA?AV?$complex@N@1@AEBV21@0@Z
-@ stub -arch=win32 ?pow@std@@YA?AV?$complex@N@1@ABV21@ABN@Z
+@ cdecl -arch=win32 ?pow@std@@YA?AV?$complex@N@1@ABNABV21@@Z(ptr ptr ptr) complex_double_pow_dc
+@ cdecl -arch=win64 ?pow@std@@YA?AV?$complex@N@1@AEBNAEBV21@@Z(ptr ptr ptr) complex_double_pow_dc
+@ cdecl -arch=win32 ?pow@std@@YA?AV?$complex@N@1@ABV21@0@Z(ptr ptr ptr) complex_double_pow
+@ cdecl -arch=win64 ?pow@std@@YA?AV?$complex@N@1@AEBV21@0@Z(ptr ptr ptr) complex_double_pow
+@ cdecl -arch=win32 ?pow@std@@YA?AV?$complex@N@1@ABV21@ABN@Z(ptr ptr ptr) complex_double_pow_cd
 @ cdecl -arch=win64 ?pow@std@@YA?AV?$complex@N@1@AEBV21@AEBN@Z(ptr ptr long) complex_double_pow_ci
 @ cdecl -arch=win32 ?pow@std@@YA?AV?$complex@N@1@ABV21@H@Z(ptr ptr long) complex_double_pow_ci
-@ stub -arch=win64 ?pow@std@@YA?AV?$complex@N@1@AEBV21@H@Z
-@ stub -arch=win32 ?pow@std@@YA?AV?$complex@O@1@ABOABV21@@Z
-@ stub -arch=win64 ?pow@std@@YA?AV?$complex@O@1@AEBOAEBV21@@Z
-@ stub -arch=win32 ?pow@std@@YA?AV?$complex@O@1@ABV21@0@Z
-@ stub -arch=win64 ?pow@std@@YA?AV?$complex@O@1@AEBV21@0@Z
-@ stub -arch=win32 ?pow@std@@YA?AV?$complex@O@1@ABV21@ABO@Z
-@ stub -arch=win64 ?pow@std@@YA?AV?$complex@O@1@AEBV21@AEBO@Z
+@ cdecl -arch=win64 ?pow@std@@YA?AV?$complex@N@1@AEBV21@H@Z(ptr ptr long) complex_double_pow_ci
+@ cdecl -arch=win32 ?pow@std@@YA?AV?$complex@O@1@ABOABV21@@Z(ptr ptr ptr) complex_double_pow_dc
+@ cdecl -arch=win64 ?pow@std@@YA?AV?$complex@O@1@AEBOAEBV21@@Z(ptr ptr ptr) complex_double_pow_dc
+@ cdecl -arch=win32 ?pow@std@@YA?AV?$complex@O@1@ABV21@0@Z(ptr ptr ptr) complex_double_pow
+@ cdecl -arch=win64 ?pow@std@@YA?AV?$complex@O@1@AEBV21@0@Z(ptr ptr ptr) complex_double_pow
+@ cdecl -arch=win32 ?pow@std@@YA?AV?$complex@O@1@ABV21@ABO@Z(ptr ptr ptr) complex_double_pow_cd
+@ cdecl -arch=win64 ?pow@std@@YA?AV?$complex@O@1@AEBV21@AEBO@Z(ptr ptr ptr) complex_double_pow_cd
 @ cdecl -arch=win32 ?pow@std@@YA?AV?$complex@O@1@ABV21@H@Z(ptr ptr long) complex_double_pow_ci
 @ cdecl -arch=win64 ?pow@std@@YA?AV?$complex@O@1@AEBV21@H@Z(ptr ptr long) complex_double_pow_ci
 @ cdecl -arch=arm ?pptr@?$basic_streambuf@DU?$char_traits@D@std@@@std@@IBAPADXZ(ptr) basic_streambuf_char_pptr
@@ -4974,29 +4974,29 @@
 @ thiscall -arch=i386 ?readsome@?$basic_istream@GU?$char_traits@G@std@@@std@@QAEHPAGH@Z(ptr ptr long) basic_istream_short_readsome
 @ cdecl -arch=win64 ?readsome@?$basic_istream@GU?$char_traits@G@std@@@std@@QEAA_JPEAG_J@Z(ptr ptr long) basic_istream_short_readsome
 @ stub -arch=arm ?real@?$_Complex_base@M@std@@QAAMABM@Z
-@ stub -arch=i386 ?real@?$_Complex_base@M@std@@QAEMABM@Z
-@ stub -arch=win64 ?real@?$_Complex_base@M@std@@QEAAMAEBM@Z
+@ thiscall -arch=i386 ?real@?$_Complex_base@M@std@@QAEMABM@Z(ptr ptr) complex_float_real_set
+@ cdecl -arch=win64 ?real@?$_Complex_base@M@std@@QEAAMAEBM@Z(ptr ptr) complex_float_real_set
 @ stub -arch=arm ?real@?$_Complex_base@M@std@@QBAMXZ
-@ stub -arch=i386 ?real@?$_Complex_base@M@std@@QBEMXZ
-@ stub -arch=win64 ?real@?$_Complex_base@M@std@@QEBAMXZ
+@ thiscall -arch=i386 ?real@?$_Complex_base@M@std@@QBEMXZ(ptr) complex_float_real_get
+@ cdecl -arch=win64 ?real@?$_Complex_base@M@std@@QEBAMXZ(ptr) complex_float_real_get
 @ stub -arch=arm ?real@?$_Complex_base@N@std@@QAANABN@Z
-@ stub -arch=i386 ?real@?$_Complex_base@N@std@@QAENABN@Z
-@ stub -arch=win64 ?real@?$_Complex_base@N@std@@QEAANAEBN@Z
+@ thiscall -arch=i386 ?real@?$_Complex_base@N@std@@QAENABN@Z(ptr ptr) complex_double_real_set
+@ cdecl -arch=win64 ?real@?$_Complex_base@N@std@@QEAANAEBN@Z(ptr ptr) complex_double_real_set
 @ stub -arch=arm ?real@?$_Complex_base@N@std@@QBANXZ
-@ stub -arch=i386 ?real@?$_Complex_base@N@std@@QBENXZ
-@ stub -arch=win64 ?real@?$_Complex_base@N@std@@QEBANXZ
+@ thiscall -arch=i386 ?real@?$_Complex_base@N@std@@QBENXZ(ptr) complex_double_real_get
+@ cdecl -arch=win64 ?real@?$_Complex_base@N@std@@QEBANXZ(ptr) complex_double_real_get
 @ stub -arch=arm ?real@?$_Complex_base@O@std@@QAAOABO@Z
-@ stub -arch=i386 ?real@?$_Complex_base@O@std@@QAEOABO@Z
-@ stub -arch=win64 ?real@?$_Complex_base@O@std@@QEAAOAEBO@Z
+@ thiscall -arch=i386 ?real@?$_Complex_base@O@std@@QAEOABO@Z(ptr ptr) complex_double_real_set
+@ cdecl -arch=win64 ?real@?$_Complex_base@O@std@@QEAAOAEBO@Z(ptr ptr) complex_double_real_set
 @ stub -arch=arm ?real@?$_Complex_base@O@std@@QBAOXZ
-@ stub -arch=i386 ?real@?$_Complex_base@O@std@@QBEOXZ
-@ stub -arch=win64 ?real@?$_Complex_base@O@std@@QEBAOXZ
-@ stub -arch=win32 ?real@std@@YAMABV?$complex@M@1@@Z
-@ stub -arch=win64 ?real@std@@YAMAEBV?$complex@M@1@@Z
-@ stub -arch=win32 ?real@std@@YANABV?$complex@N@1@@Z
-@ stub -arch=win64 ?real@std@@YANAEBV?$complex@N@1@@Z
-@ stub -arch=win32 ?real@std@@YAOABV?$complex@O@1@@Z
-@ stub -arch=win64 ?real@std@@YAOAEBV?$complex@O@1@@Z
+@ thiscall -arch=i386 ?real@?$_Complex_base@O@std@@QBEOXZ(ptr) complex_double_real_get
+@ cdecl -arch=win64 ?real@?$_Complex_base@O@std@@QEBAOXZ(ptr) complex_double_real_get
+@ cdecl -arch=win32 ?real@std@@YAMABV?$complex@M@1@@Z(ptr) complex_float_real
+@ cdecl -arch=win64 ?real@std@@YAMAEBV?$complex@M@1@@Z(ptr) complex_float_real
+@ cdecl -arch=win32 ?real@std@@YANABV?$complex@N@1@@Z(ptr) complex_double_real
+@ cdecl -arch=win64 ?real@std@@YANAEBV?$complex@N@1@@Z(ptr) complex_double_real
+@ cdecl -arch=win32 ?real@std@@YAOABV?$complex@O@1@@Z(ptr) complex_double_real
+@ cdecl -arch=win64 ?real@std@@YAOAEBV?$complex@O@1@@Z(ptr) complex_double_real
 @ cdecl -arch=arm ?register_callback@ios_base@std@@QAAXP6AXW4event@12@AAV12@H@ZH@Z(ptr ptr long) ios_base_register_callback
 @ thiscall -arch=i386 ?register_callback@ios_base@std@@QAEXP6AXW4event@12@AAV12@H@ZH@Z(ptr ptr long) ios_base_register_callback
 @ cdecl -arch=win64 ?register_callback@ios_base@std@@QEAAXP6AXW4event@12@AEAV12@H@ZH@Z(ptr ptr long) ios_base_register_callback
@@ -5303,21 +5303,21 @@
 @ cdecl ?signaling_NaN@?$numeric_limits@N@std@@SANXZ() std_numeric_limits_double_signaling_NaN
 @ cdecl ?signaling_NaN@?$numeric_limits@O@std@@SAOXZ() std_numeric_limits_long_double_signaling_NaN
 @ cdecl ?signaling_NaN@?$numeric_limits@_N@std@@SA_NXZ() std_numeric_limits_bool_signaling_NaN
-@ stub ?sin@?$_Ctr@M@std@@SAMM@Z
-@ stub ?sin@?$_Ctr@N@std@@SANN@Z
-@ stub ?sin@?$_Ctr@O@std@@SAOO@Z
-@ stub -arch=win32 ?sin@std@@YA?AV?$complex@M@1@ABV21@@Z
-@ stub -arch=win64 ?sin@std@@YA?AV?$complex@M@1@AEBV21@@Z
-@ stub -arch=win32 ?sin@std@@YA?AV?$complex@N@1@ABV21@@Z
-@ stub -arch=win64 ?sin@std@@YA?AV?$complex@N@1@AEBV21@@Z
-@ stub -arch=win32 ?sin@std@@YA?AV?$complex@O@1@ABV21@@Z
-@ stub -arch=win64 ?sin@std@@YA?AV?$complex@O@1@AEBV21@@Z
-@ stub -arch=win32 ?sinh@std@@YA?AV?$complex@M@1@ABV21@@Z
-@ stub -arch=win64 ?sinh@std@@YA?AV?$complex@M@1@AEBV21@@Z
-@ stub -arch=win32 ?sinh@std@@YA?AV?$complex@N@1@ABV21@@Z
-@ stub -arch=win64 ?sinh@std@@YA?AV?$complex@N@1@AEBV21@@Z
-@ stub -arch=win32 ?sinh@std@@YA?AV?$complex@O@1@ABV21@@Z
-@ stub -arch=win64 ?sinh@std@@YA?AV?$complex@O@1@AEBV21@@Z
+@ cdecl ?sin@?$_Ctr@M@std@@SAMM@Z(float) std_Ctraits_float_sin
+@ cdecl ?sin@?$_Ctr@N@std@@SANN@Z(double) std_Ctraits_double_sin
+@ cdecl ?sin@?$_Ctr@O@std@@SAOO@Z(double) std_Ctraits_long_double_sin
+@ cdecl -arch=win32 ?sin@std@@YA?AV?$complex@M@1@ABV21@@Z(ptr ptr) complex_float_sin
+@ cdecl -arch=win64 ?sin@std@@YA?AV?$complex@M@1@AEBV21@@Z(ptr ptr) complex_float_sin
+@ cdecl -arch=win32 ?sin@std@@YA?AV?$complex@N@1@ABV21@@Z(ptr ptr) complex_double_sin
+@ cdecl -arch=win64 ?sin@std@@YA?AV?$complex@N@1@AEBV21@@Z(ptr ptr) complex_double_sin
+@ cdecl -arch=win32 ?sin@std@@YA?AV?$complex@O@1@ABV21@@Z(ptr ptr) complex_double_sin
+@ cdecl -arch=win64 ?sin@std@@YA?AV?$complex@O@1@AEBV21@@Z(ptr ptr) complex_double_sin
+@ cdecl -arch=win32 ?sinh@std@@YA?AV?$complex@M@1@ABV21@@Z(ptr ptr) complex_float_sinh
+@ cdecl -arch=win64 ?sinh@std@@YA?AV?$complex@M@1@AEBV21@@Z(ptr ptr) complex_float_sinh
+@ cdecl -arch=win32 ?sinh@std@@YA?AV?$complex@N@1@ABV21@@Z(ptr ptr) complex_double_sinh
+@ cdecl -arch=win64 ?sinh@std@@YA?AV?$complex@N@1@AEBV21@@Z(ptr ptr) complex_double_sinh
+@ cdecl -arch=win32 ?sinh@std@@YA?AV?$complex@O@1@ABV21@@Z(ptr ptr) complex_double_sinh
+@ cdecl -arch=win64 ?sinh@std@@YA?AV?$complex@O@1@AEBV21@@Z(ptr ptr) complex_double_sinh
 @ cdecl -arch=arm ?size@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@QBAIXZ(ptr) MSVCP_basic_string_char_length
 @ thiscall -arch=i386 ?size@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@QBEIXZ(ptr) MSVCP_basic_string_char_length
 @ cdecl -arch=win64 ?size@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@QEBA_KXZ(ptr) MSVCP_basic_string_char_length
@@ -5348,15 +5348,15 @@
 @ cdecl -arch=arm ?sputn@?$basic_streambuf@GU?$char_traits@G@std@@@std@@QAAHPBGH@Z(ptr ptr long) basic_streambuf_wchar_sputn
 @ thiscall -arch=i386 ?sputn@?$basic_streambuf@GU?$char_traits@G@std@@@std@@QAEHPBGH@Z(ptr ptr long) basic_streambuf_wchar_sputn
 @ cdecl -arch=win64 ?sputn@?$basic_streambuf@GU?$char_traits@G@std@@@std@@QEAA_JPEBG_J@Z(ptr ptr long) basic_streambuf_wchar_sputn
-@ stub ?sqrt@?$_Ctr@M@std@@SAMM@Z
-@ stub ?sqrt@?$_Ctr@N@std@@SANN@Z
-@ stub ?sqrt@?$_Ctr@O@std@@SAOO@Z
-@ stub -arch=win32 ?sqrt@std@@YA?AV?$complex@M@1@ABV21@@Z
-@ stub -arch=win64 ?sqrt@std@@YA?AV?$complex@M@1@AEBV21@@Z
-@ stub -arch=win32 ?sqrt@std@@YA?AV?$complex@N@1@ABV21@@Z
-@ stub -arch=win64 ?sqrt@std@@YA?AV?$complex@N@1@AEBV21@@Z
-@ stub -arch=win32 ?sqrt@std@@YA?AV?$complex@O@1@ABV21@@Z
-@ stub -arch=win64 ?sqrt@std@@YA?AV?$complex@O@1@AEBV21@@Z
+@ cdecl ?sqrt@?$_Ctr@M@std@@SAMM@Z(float) std_Ctraits_float_sqrt
+@ cdecl ?sqrt@?$_Ctr@N@std@@SANN@Z(double) std_Ctraits_double_sqrt
+@ cdecl ?sqrt@?$_Ctr@O@std@@SAOO@Z(double) std_Ctraits_long_double_sqrt
+@ cdecl -arch=win32 ?sqrt@std@@YA?AV?$complex@M@1@ABV21@@Z(ptr ptr) complex_float_sqrt
+@ cdecl -arch=win64 ?sqrt@std@@YA?AV?$complex@M@1@AEBV21@@Z(ptr ptr) complex_float_sqrt
+@ cdecl -arch=win32 ?sqrt@std@@YA?AV?$complex@N@1@ABV21@@Z(ptr ptr) complex_double_sqrt
+@ cdecl -arch=win64 ?sqrt@std@@YA?AV?$complex@N@1@AEBV21@@Z(ptr ptr) complex_double_sqrt
+@ cdecl -arch=win32 ?sqrt@std@@YA?AV?$complex@O@1@ABV21@@Z(ptr ptr) complex_double_sqrt
+@ cdecl -arch=win64 ?sqrt@std@@YA?AV?$complex@O@1@AEBV21@@Z(ptr ptr) complex_double_sqrt
 @ cdecl -arch=arm ?stossc@?$basic_streambuf@DU?$char_traits@D@std@@@std@@QAAXXZ(ptr) basic_streambuf_char_stossc
 @ thiscall -arch=i386 ?stossc@?$basic_streambuf@DU?$char_traits@D@std@@@std@@QAEXXZ(ptr) basic_streambuf_char_stossc
 @ cdecl -arch=win64 ?stossc@?$basic_streambuf@DU?$char_traits@D@std@@@std@@QEAAXXZ(ptr) basic_streambuf_char_stossc
diff --git a/dlls/msvcp90/math.c b/dlls/msvcp90/math.c
index a502692..3c00104 100644
--- a/dlls/msvcp90/math.c
+++ b/dlls/msvcp90/math.c
@@ -2443,3 +2443,22 @@ float __cdecl std__XLgamma_float(float z)
 {
     return std__XLgamma_double(z);
 }
+
+#if _MSVCP_VER < 70
+/* msvcp60's compound assignments are function templates with two complex
+ * arguments: ??$?XMM@std@@YAAAV?$complex@M@0@AAV10@ABV10@@Z and the others */
+complex_float* __cdecl complex_float_mult_assign_tmpl(complex_float *l, const complex_float *r) { return complex_float_mult_assign(l, r); }
+complex_float* __cdecl complex_float_add_assign_tmpl(complex_float *l, const complex_float *r) { return complex_float_add_assign(l, r); }
+complex_float* __cdecl complex_float_sub_assign_tmpl(complex_float *l, const complex_float *r)
+{
+#ifdef SG_MUTANT_COMPLEX60_SUB_IS_ADD
+    return complex_float_add_assign(l, r);
+#endif
+    return complex_float_sub_assign(l, r);
+}
+complex_float* __cdecl complex_float_div_assign_tmpl(complex_float *l, const complex_float *r) { return complex_float_div_assign(l, r); }
+complex_double* __cdecl complex_double_mult_assign_tmpl(complex_double *l, const complex_double *r) { return complex_double_mult_assign(l, r); }
+complex_double* __cdecl complex_double_add_assign_tmpl(complex_double *l, const complex_double *r) { return complex_double_add_assign(l, r); }
+complex_double* __cdecl complex_double_sub_assign_tmpl(complex_double *l, const complex_double *r) { return complex_double_sub_assign(l, r); }
+complex_double* __cdecl complex_double_div_assign_tmpl(complex_double *l, const complex_double *r) { return complex_double_div_assign(l, r); }
+#endif
